{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Header conflict protein.fasta and CDS.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " - While CDS.fasta files have sequential headers e.g. main1, main2 etc, the protein.fasta header produced by gffread is not sequential.\n",
    " - To ensure that the protein prediction can be linked to CDS ID, use maker scripts to change header within gff then remake protein.fasta and CDS.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://weatherby.genetics.utah.edu/MAKER/wiki/index.php/MAKER_Tutorial_for_WGS_Assembly_and_Annotation_Winter_School_2018 #rename gene models and other data."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The script maker_map_ids produces a map file of original header paired to sequential headers.\n",
    "The script map_gff_ids takes the map file and renames the gff file."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The scripts were copied from /bioinf_resources/maker/bin"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Urgent update"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " - Find the CDS input you used to extract protein from that gave v100_protein.fa which was used for blast2go.\n",
    " - Run BUSCO checks on CDS generated by unique region."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Troubleshooting"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " - The commands used to translate CDS to proteins can be found in f307/merged/annotation/pipeline/protein_annotation.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotation=\"/home/data/bioinf_training/colin/comparative_genomics/fusarium_decemcellulares/merged/annotation/pipeline/3_callgenes/maker_genome/FD.maker.output/FD.all.gff\"\n",
    "genomic=\"/home/data/bioinf_training/colin/comparative_genomics/fusarium_decemcellulares/merged/data/F_decem_v100.fasta\"\n",
    "gffread $annotation -g $genomic -y v100_protein.fa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Next steps"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " - Use the maker scripts to change sequence id in both f307 and f200 unique region's gff output.\n",
    " - Use the same procedures to extract protein sequence from the unique regions.\n",
    " - Run blast2go on a concatenated protein.fasta from both f307 and f200.\n",
    " - Extract CDS again using the renamed gff for both f200 and f307.\n",
    " - Use sort.py script to find only the CDS sequences with start and stop codons for CDS in maker_genome directory.\n",
    " - Run BUSCO checks on CDS generated by unique region.\n",
    " - Tidy your directories and archive or delete useless files."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Renaming the gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "perl maker_map_ids --prefix f307MAIN_ --justify 8 FD.all.gff > map\n",
    "perl map_gff_ids map  FD.all.gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "perl maker_map_ids --prefix f200UNIQUE_ --justify 8 FD.all.gff > map\n",
    "perl map_gff_ids map  FD.all.gff"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CDS filtering"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- The number of CDS sequences starting with a start codon AND  ending with a stop codon:\n",
    "    - 17441 CDS sequences for f307   \n",
    "    - 675 CDS sequences for f200.\n",
    "\n",
    "- The number of CDS sequences either missing a start codon, a stop codon or both:\n",
    "    - 32 CDS sequences for f307\n",
    "    - 116 CDS sequences for f200."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pangenome\n",
    "\n",
    " - After renaming the 2 gff files, the standard procedure of extract CDS, protein, sort.py were used.\n",
    " - Pangeome was constructed by concatenating the filtered CDS from the 2 isolates."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
