{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import requests\n",
    "import sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "glist=\"T:/colin/BRice/V3/genepage-insight/example_list.txt\"\n",
    "klist=\"T:/colin/BRice/V3/genepage-insight/mock_keyword_list.txt\"\n",
    "olist=\"T:/colin/BRice/V3/genepage-insight/example_list/example_list.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "def append_summary():\n",
    "    with open(klist, \"r\") as fk:\n",
    "        pheno=[]\n",
    "        for line in fk:\n",
    "            pheno.append(line.rstrip())\n",
    "        #print(pheno)\n",
    "        summary=pd.read_csv(glist, sep=\"\\t\", header=None)\n",
    "        genes=list(summary[0])\n",
    "        #return summary\n",
    "        \n",
    "\n",
    "        #creating knetminer genepage urls.\n",
    "        network_view=[]\n",
    "        keyw2 = \"+OR+\".join(\"({})\".format(i.replace(\" \", \"+AND+\")) for i in pheno)\n",
    "        #print(keyw2)\n",
    "        #define species\n",
    "        species=\"wheatknet\"\n",
    "        for i in genes:\n",
    "            link=\"http://knetminer.rothamsted.ac.uk/{}/genepage?list={}&keyword={}\".format(species, i, keyw2)\n",
    "            r=requests.get(link)\n",
    "            network_view.append(r.url)\n",
    "\n",
    "        #obtaining knetscores for genes\n",
    "        keyw1 = \"%20OR%20\".join(\"({})\".format(i.replace(\" \", \"+AND+\")) for i in pheno)\n",
    "        genestr=(\",\").join(genes)\n",
    "        link=\"http://knetminer.rothamsted.ac.uk/{}/genome?\".format(species)\n",
    "        parameters={\"keyword\":keyw1, \"list\":genestr}\n",
    "        r=requests.get(link, params=parameters)\n",
    "\n",
    "        #check if requests is successful\n",
    "        if not r.ok:\n",
    "                r.raise_for_status()\n",
    "                sys.exit()\n",
    "        \n",
    "        #extract unicode string of geneTable decoded from json\n",
    "        decoded=r.json()[u'geneTable'].split(\"\\t\")\n",
    "        #remove space or newline at the end\n",
    "        decoded=(decoded)[:-1]\n",
    "        \n",
    "        colnum=9\n",
    "        #tabulate genetable into 9 columns.\n",
    "        genetable=np.array(decoded).reshape(len(decoded)//colnum, colnum)\n",
    "        genetable=pd.DataFrame(genetable[1:,:], columns=genetable[0,:])\n",
    "                \n",
    "        knetgenes=list(genetable[u'ACCESSION'])\n",
    "        knetscores=list(genetable[u'SCORE'])\n",
    "        knetchro=list(genetable[u'CHRO'])\n",
    "        knetstart=list(genetable[u'START'])\n",
    "\n",
    "        #map genes to snps via a dictionary.\n",
    "        knetdict=dict(zip(knetgenes, knetscores))\n",
    "        ordered_score=[]\n",
    "        for i in summary[0]:\n",
    "            #convert gene id to upper case to avoid sensitivity issues.\n",
    "            i=i.upper()\n",
    "            ordered_score.append(knetdict[u'{}'.format(i)])\n",
    "        summary[u'knetscore'] = ordered_score\n",
    "        summary[u'network_view']=network_view\n",
    "        summary.to_csv(olist, sep=\"\\t\")\n",
    "      \n",
    "        return summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>knetscore</th>\n",
       "      <th>network_view</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRAESCS1A02G173400</td>\n",
       "      <td>306.22</td>\n",
       "      <td>https://knetminer.rothamsted.ac.uk/wheatknet/g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRAESCS4B02G293600</td>\n",
       "      <td>13.10</td>\n",
       "      <td>https://knetminer.rothamsted.ac.uk/wheatknet/g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRAESCS6A02G024400</td>\n",
       "      <td>0.00</td>\n",
       "      <td>https://knetminer.rothamsted.ac.uk/wheatknet/g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRAESCS6A02G222000</td>\n",
       "      <td>104.23</td>\n",
       "      <td>https://knetminer.rothamsted.ac.uk/wheatknet/g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRAESCS5A02G030800</td>\n",
       "      <td>227.47</td>\n",
       "      <td>https://knetminer.rothamsted.ac.uk/wheatknet/g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TRAESCS4D02G275500</td>\n",
       "      <td>40.21</td>\n",
       "      <td>https://knetminer.rothamsted.ac.uk/wheatknet/g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TRAESCS7D02G538400</td>\n",
       "      <td>76.91</td>\n",
       "      <td>https://knetminer.rothamsted.ac.uk/wheatknet/g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>TRAESCS7A02G107400</td>\n",
       "      <td>88.59</td>\n",
       "      <td>https://knetminer.rothamsted.ac.uk/wheatknet/g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>TRAESCS3A02G231900</td>\n",
       "      <td>0.00</td>\n",
       "      <td>https://knetminer.rothamsted.ac.uk/wheatknet/g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>TRAESCS3B02G302100</td>\n",
       "      <td>35.66</td>\n",
       "      <td>https://knetminer.rothamsted.ac.uk/wheatknet/g...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    0 knetscore  \\\n",
       "0  TRAESCS1A02G173400    306.22   \n",
       "1  TRAESCS4B02G293600     13.10   \n",
       "2  TRAESCS6A02G024400      0.00   \n",
       "3  TRAESCS6A02G222000    104.23   \n",
       "4  TRAESCS5A02G030800    227.47   \n",
       "5  TRAESCS4D02G275500     40.21   \n",
       "6  TRAESCS7D02G538400     76.91   \n",
       "7  TRAESCS7A02G107400     88.59   \n",
       "8  TRAESCS3A02G231900      0.00   \n",
       "9  TRAESCS3B02G302100     35.66   \n",
       "\n",
       "                                        network_view  \n",
       "0  https://knetminer.rothamsted.ac.uk/wheatknet/g...  \n",
       "1  https://knetminer.rothamsted.ac.uk/wheatknet/g...  \n",
       "2  https://knetminer.rothamsted.ac.uk/wheatknet/g...  \n",
       "3  https://knetminer.rothamsted.ac.uk/wheatknet/g...  \n",
       "4  https://knetminer.rothamsted.ac.uk/wheatknet/g...  \n",
       "5  https://knetminer.rothamsted.ac.uk/wheatknet/g...  \n",
       "6  https://knetminer.rothamsted.ac.uk/wheatknet/g...  \n",
       "7  https://knetminer.rothamsted.ac.uk/wheatknet/g...  \n",
       "8  https://knetminer.rothamsted.ac.uk/wheatknet/g...  \n",
       "9  https://knetminer.rothamsted.ac.uk/wheatknet/g...  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "append_summary()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "def snippet():\n",
    "    with open(klist, \"r\") as fk:\n",
    "        pheno=[]\n",
    "        for line in fk:\n",
    "            pheno.append(line.rstrip())\n",
    "        #print(pheno)\n",
    "        summary=pd.read_csv(glist, sep=\"\\t\", header=None)\n",
    "        summary.rename (\n",
    "        columns={\n",
    "            0:\"GENE\"\n",
    "        }, inplace=True\n",
    "        )\n",
    "        genes=list(summary[\"GENE\"])\n",
    "        \n",
    "        #creating knetminer genepage urls.\n",
    "        network_view=[]\n",
    "        keyw2 = \"+OR+\".join(\"({})\".format(i.replace(\" \", \"+AND+\")) for i in pheno)\n",
    "        #print(keyw2)\n",
    "        #define species\n",
    "        species=\"wheatknet\"\n",
    "        for i in genes:\n",
    "            link=\"http://knetminer.rothamsted.ac.uk/{}/genepage?list={}&keyword={}\".format(species, i, keyw2)\n",
    "            r=requests.get(link)\n",
    "            network_view.append(r.url)\n",
    "\n",
    "        #obtaining knetscores for genes\n",
    "        keyw1 = \"%20OR%20\".join(\"({})\".format(i.replace(\" \", \"+AND+\")) for i in pheno)\n",
    "        genestr=(\",\").join(genes)\n",
    "        link=\"http://knetminer.rothamsted.ac.uk/{}/genome?\".format(species)\n",
    "        parameters={\"keyword\":keyw1, \"list\":genestr}\n",
    "        r=requests.get(link, params=parameters)\n",
    "        url = \"http://knetminer.rothamsted.ac.uk/{}/genome?keyword={}&list={}\".format(species, keyw1, genestr)\n",
    "        return url\n",
    "\n",
    "        #check if requests is successful\n",
    "        if not r.ok:\n",
    "                r.raise_for_status()\n",
    "                sys.exit()\n",
    "        \n",
    "        #extract unicode string of geneTable decoded from json\n",
    "        decoded=r.json()[u'geneTable'].split(\"\\t\")\n",
    "        #remove space or newline at the end\n",
    "        decoded=(decoded)[:-1]\n",
    "        \n",
    "        colnum=9\n",
    "        #tabulate genetable into 9 columns.\n",
    "        genetable=np.array(decoded).reshape(len(decoded)//colnum, colnum)\n",
    "        genetable=pd.DataFrame(genetable[1:,:], columns=genetable[0,:])\n",
    "        return genetable.shape\n",
    "                \n",
    "        knetgenes=list(genetable[u'ACCESSION'])\n",
    "        knetscores=list(genetable[u'SCORE'])\n",
    "        knetchro=list(genetable[u'CHRO'])\n",
    "        knetstart=list(genetable[u'START'])\n",
    "        #map genes to snps via a dictionary.\n",
    "        knetdict=dict(zip(knetgenes, knetscores))\n",
    "        print knetdict\n",
    "        ordered_score=[]\n",
    "        for i in genes:\n",
    "            #convert gene id to upper case to avoid sensitivity issues.\n",
    "            i=i.upper()\n",
    "            ordered_score.append(knetdict[u'{}'.format(i)])\n",
    "        summary[u'knetscore'] = ordered_score\n",
    "        summary[u'network_view']=network_view\n",
    "        summary.to_csv(olist, sep=\"\\t\", index=False)\n",
    "        \n",
    "        summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'http://knetminer.rothamsted.ac.uk/wheatknet/genome?keyword=(Germination)%20OR%20(Coleoptile+AND+length)%20OR%20(Mesocotyl+AND+length)&list=TRAESCS1A02G173400,TRAESCS4B02G293600,TRAESCS6A02G024400,TRAESCS6A02G222000,TRAESCS5A02G030800'"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snippet()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"TRAESCS1A02G173400\" == u'TRAESCS1A02G173400'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "dict={'TRAESCS1A02G173400': '306.22', \n",
    "      'TRAESCS5A02G030800': '227.47', \n",
    "      'TRAESCS6A02G222000': '104.23', \n",
    "      'TRAESCS4B02G293600': '13.10', \n",
    "      'TRAESCS6A02G024400': '0.00'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "list=[\"TRAESCS1A02G173400\", \"TRAESCS5A02G030800\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "306.22\n",
      "227.47\n"
     ]
    }
   ],
   "source": [
    "for i in list:\n",
    "    print(dict[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
