{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# updated repository readME using the following template\n",
    "\n",
    "https://gist.github.com/PurpleBooth/109311bb0361f32d87a2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Experimenting to extract all species from ensembl GET /info/ species\n",
    "\n",
    "This is assuming such a list doesn't already exist or is relevant to rest server."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The only keys in the decoded content are:\n",
      "species\n",
      "the value of species key is a\n",
      "<type 'list'>\n",
      "\n",
      "\n",
      "inspect one element of all species list\n",
      "{u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'AnoCar2.0', u'name': u'anolis_carolinensis', u'common_name': u'green anole', u'accession': u'GCA_000090745.1', u'taxon_id': u'28377', u'strain': None, u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'release': 96, u'display_name': u'Anole lizard', u'aliases': [u'anoliscarolinensis', u'acar', u'green anole', u'anolis', u'anolis carolinensis', u'acarolinensis', u'anole lizard', u'anolis_lizard', u'28377', u'anocar', u'green anole lizard']}\n",
      "extracting all the species common name and ensembl name!\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'\\nfor i in allsp:\\n    ensemblname = str(i[u\\'name\\'])\\n    commonname = str(i[u\\'common_name\\'])\\n    spacer = \"\\t\" * 3\\n    print(\"{}{}{}\".format(commonname, spacer, ensemblname))\\n'"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import requests, sys\n",
    " \n",
    "server = \"https://rest.ensembl.org\"\n",
    "ext = \"/info/species?\"\n",
    " \n",
    "r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"})\n",
    " \n",
    "if not r.ok:\n",
    "  r.raise_for_status()\n",
    "  sys.exit()\n",
    " \n",
    "decoded = r.json()\n",
    "\n",
    "#print all keys. There is only one key in decoded, species.\n",
    "for key in decoded:\n",
    "    print(\"The only keys in the decoded content are:\")\n",
    "    print(key)\n",
    "\n",
    "allsp = decoded[u\"species\"]\n",
    "print(\"the value of species key is a\")\n",
    "print(type(allsp))\n",
    "print(\"\\n\")\n",
    "\n",
    "\n",
    "print(\"inspect one element of all species list\")\n",
    "print(allsp[0])\n",
    "\n",
    "\n",
    "print(\"extracting all the species common name and ensembl name!\")\n",
    "'''\n",
    "for i in allsp:\n",
    "    ensemblname = str(i[u'name'])\n",
    "    commonname = str(i[u'common_name'])\n",
    "    spacer = \"\\t\" * 3\n",
    "    print(\"{}{}{}\".format(commonname, spacer, ensemblname))\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<type 'dict'>\n"
     ]
    }
   ],
   "source": [
    "print(type(decoded))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "species\n"
     ]
    }
   ],
   "source": [
    "for key in decoded:\n",
    "    print(key)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<type 'list'>\n"
     ]
    }
   ],
   "source": [
    "allsp = decoded[u'species']\n",
    "print(type(allsp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[{u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'display_name': u'Mouse NZO/HlLtJ', u'name': u'mus_musculus_nzohlltj', u'strain': u'NZO/HlLtJ', u'common_name': u'house mouse', u'accession': u'GCA_001624745.1', u'release': 96, u'groups': [u'funcgen', u'core'], u'taxon_id': u'10090', u'assembly': u'NZO_HlLtJ_v1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Little spotted kiwi', u'name': u'apteryx_owenii', u'strain': None, u'accession': u'GCA_003342965.1', u'taxon_id': u'8824', u'release': 96, u'groups': [u'rnaseq', u'core'], u'common_name': u'little spotted kiwi', u'assembly': u'aptOwe1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Chrysemys_picta_bellii-3.0.3', u'name': u'chrysemys_picta_bellii', u'common_name': u'Western painted turtle', u'accession': u'GCA_000241765.2', u'taxon_id': u'8478', u'strain': None, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'release': 96, u'display_name': u'Painted turtle', u'aliases': [u'western painted turtle', u'painted turtle']}, {u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'display_name': u'Mouse', u'name': u'mus_musculus', u'strain': u'reference (CL57BL6)', u'accession': u'GCA_000001635.8', u'taxon_id': u'10090', u'common_name': u'house mouse', u'groups': [u'variation', u'cdna', u'rnaseq', u'funcgen', u'core', u'otherfeatures'], u'release': 96, u'assembly': u'GRCm38', u'aliases': [u'musmus', u'm_musculus', u'house mouse', u'mmus', u'10090', u'mus musculus', u'mmusculus', u'mouse', u'ensmm', u'mus']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'ASM275486v1', u'name': u'callithrix_jacchus', u'common_name': u'white-tufted-ear marmoset', u'accession': u'GCA_002754865.1', u'strain': None, u'groups': [u'rnaseq', u'core', u'funcgen'], u'taxon_id': u'9483', u'display_name': u'Marmoset', u'aliases': [u'9483', u'cjacchus321', u'cjac', u'marmoset', u'caljac', u'cj321', u'callithrix jacchus', u'cjacchus', u'white-tufted-ear marmoset'], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'LepOcu1', u'name': u'lepisosteus_oculatus', u'display_name': u'Spotted gar', u'accession': u'GCA_000242695.1', u'taxon_id': u'7918', u'strain': None, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'common_name': u'Spotted gar', u'release': 96, u'aliases': [u'7918', u'loculatus', u'lepocu', u'locu', u'spotted gar', u'lepisosteus oculatus']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Tasmanian devil', u'name': u'sarcophilus_harrisii', u'strain': None, u'common_name': u'Tasmanian devil', u'accession': u'GCA_000189315.1', u'release': 96, u'groups': [u'core', u'otherfeatures'], u'taxon_id': u'9305', u'assembly': u'DEVIL7.0', u'aliases': [u'tasmanian_devil', u'devil', u'sarcophilus harrisii', u'9305', u'tdevil', u'sharrisii', u'shar', u'sarhar', u'tasmanian devil']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Fundulus_heteroclitus-3.0.2', u'name': u'fundulus_heteroclitus', u'common_name': u'mummichog', u'accession': u'GCA_000826765.1', u'taxon_id': u'8078', u'strain': None, u'groups': [u'rnaseq', u'core', u'otherfeatures', u'funcgen'], u'release': 96, u'display_name': u'Mummichog', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Ocean sunfish', u'name': u'mola_mola', u'strain': None, u'accession': u'GCA_001698575.1', u'taxon_id': u'94237', u'common_name': u'ocean sunfish', u'groups': [u'core'], u'release': 96, u'assembly': u'ASM169857v1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'CAROLI_EIJ_v1.1', u'name': u'mus_caroli', u'accession': u'GCA_900094665.2', u'taxon_id': u'10089', u'strain': u'CAROLI_EIJ', u'groups': [u'core'], u'common_name': u'Ryukyu mouse', u'display_name': u'Ryukyu mouse', u'aliases': [u'mcaroli', u'10089', u'muscar', u'mus caroli', u'ryukyu mouse', u'mcar'], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'display_name': u'Mouse FVB/NJ', u'name': u'mus_musculus_fvbnj', u'strain': u'FVB/NJ', u'common_name': u'house mouse', u'accession': u'GCA_001624535.1', u'release': 96, u'groups': [u'core', u'funcgen'], u'taxon_id': u'10090', u'assembly': u'FVB_NJ_v1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Monterrey platyfish', u'name': u'xiphophorus_couchianus', u'strain': None, u'accession': u'GCA_001444195.1', u'common_name': u'Monterrey platyfish', u'assembly': u'Xiphophorus_couchianus-4.0.1', u'groups': [u'core'], u'taxon_id': u'32473', u'release': 96, u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'PanPar1.0', u'name': u'panthera_pardus', u'display_name': u'Leopard', u'common_name': u'leopard', u'accession': u'GCA_001857705.1', u'strain': None, u'groups': [u'core', u'otherfeatures'], u'taxon_id': u'9691', u'release': 96, u'aliases': [u'leopard']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Lesser hedgehog tenrec', u'name': u'echinops_telfairi', u'strain': None, u'accession': None, u'common_name': u'small Madagascar hedgehog', u'groups': [u'core'], u'taxon_id': u'9371', u'assembly': u'TENREC', u'aliases': [u'echtel', u'lesser hedgehog tenrec', u'lesser_hedgehog', u'small madagascar hedgehog', u'tenrec', u'madagascar_hedgehog', u'etel', u'echinops telfairi', u'etelfairi', u'9371'], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Microbat', u'name': u'myotis_lucifugus', u'strain': None, u'accession': u'GCA_000147115.1', u'taxon_id': u'59463', u'release': 96, u'groups': [u'core', u'otherfeatures'], u'common_name': u'little brown bat', u'assembly': u'Myoluc2.0', u'aliases': [u'mlucifugus', u'59463', u'myotis lucifugus', u'microbat', u'little_brown_bat', u'myoluc', u'little brown bat', u'mluc']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'ASM334442v1', u'name': u'ursus_americanus', u'taxon_id': u'9643', u'accession': u'GCA_003344425.1', u'release': 96, u'strain': None, u'groups': [u'rnaseq', u'core'], u'common_name': u'American black bear', u'display_name': u'American black bear', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'loxAfr3', u'name': u'loxodonta_africana', u'display_name': u'Elephant', u'common_name': u'African savanna elephant', u'accession': u'GCA_000001905.1', u'strain': None, u'groups': [u'core'], u'taxon_id': u'9785', u'release': 96, u'aliases': [u'lafricana', u'lafr', u'loxodonta africana', u'loxafr', u'african_elephant', u'elephant', u'9785', u'african savanna elephant']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'vicPac1', u'name': u'vicugna_pacos', u'accession': None, u'taxon_id': u'30538', u'strain': None, u'groups': [u'core'], u'common_name': u'alpaca', u'display_name': u'Alpaca', u'aliases': [u'vicugnapacos', u'30538', u'vicpac', u'vpacos', u'vicugna pacos', u'alpaca', u'vpac'], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'GRCz11', u'name': u'danio_rerio', u'common_name': u'zebrafish', u'accession': u'GCA_000002035.4', u'strain': None, u'groups': [u'variation', u'rnaseq', u'funcgen', u'otherfeatures', u'core'], u'taxon_id': u'7955', u'display_name': u'Zebrafish', u'aliases': [u'drer', u'danio', u'danrer', u'zfish', u'drerio', u'zebrafish', u'd_rerio', u'7955', u'danio rerio'], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'assembly': u'PWK_PhJ_v1', u'name': u'mus_musculus_pwkphj', u'display_name': u'Mouse PWK/PhJ', u'common_name': u'eastern european house mouse', u'accession': u'GCA_001624775.1', u'strain': u'PWK/PhJ', u'groups': [u'core', u'funcgen'], u'taxon_id': u'39442', u'release': 96, u'aliases': [u'mmusculus_musculus_pwkphj', u'39442', u'mus musculus musculus pwkphj', u'mus_musculus_musculus_pwkphj', u'eastern european house mouse', u'mouse pwkphj']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'OchPri2.0-Ens', u'name': u'ochotona_princeps', u'accession': None, u'taxon_id': u'9978', u'strain': None, u'groups': [u'core'], u'common_name': u'American pika', u'display_name': u'Pika', u'aliases': [u'pika', u'ochotona princeps', u'opri', u'american pika', u'ochpri', u'9978', u'oprinceps'], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'ASM369795v1', u'name': u'calidris_pygmaea', u'common_name': u'Spoon-billed sandpiper', u'accession': u'GCA_003697955.1', u'taxon_id': u'425635', u'strain': None, u'groups': [u'core', u'rnaseq'], u'release': 96, u'display_name': u'Spoon-billed sandpiper', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Chinese hamster CHOK1GS', u'name': u'cricetulus_griseus_chok1gshd', u'strain': None, u'common_name': u'Chinese hamster', u'accession': u'GCA_900186095.1', u'release': 96, u'groups': [u'rnaseq', u'core', u'otherfeatures', u'funcgen'], u'taxon_id': u'10029', u'assembly': u'CHOK1GS_HDv1', u'aliases': [u'cgriseus_chok1gshd', u'chinese hamster chok1gs', u'cgri', u'crigri', u'cricetulus griseus chok1gshd']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'PunNye1.0', u'name': u'pundamilia_nyererei', u'common_name': u'Makobe Island cichlid', u'accession': u'GCA_000239375.1', u'taxon_id': u'303518', u'strain': None, u'groups': [u'core', u'otherfeatures', u'rnaseq'], u'release': 96, u'display_name': u'Makobe Island cichlid', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'NNQGG.v01', u'name': u'neovison_vison', u'display_name': u'American mink', u'common_name': u'American mink', u'accession': u'GCA_900108605.1', u'strain': None, u'groups': [u'core', u'rnaseq'], u'taxon_id': u'452646', u'release': 96, u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'PoeFor_5.1.2', u'name': u'poecilia_formosa', u'common_name': u'Amazon molly', u'accession': u'GCA_000485575.1', u'taxon_id': u'48698', u'strain': None, u'groups': [u'core', u'otherfeatures', u'rnaseq'], u'release': 96, u'display_name': u'Amazon molly', u'aliases': [u'poefor', u'amazon molly', u'48698', u'pformosa', u'pfor', u'poecilia formosa']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Wallaby', u'name': u'notamacropus_eugenii', u'strain': None, u'accession': u'GCA_000004035.1', u'taxon_id': u'9315', u'release': 96, u'groups': [u'core'], u'common_name': u'tammar wallaby', u'assembly': u'Meug_1.0', u'aliases': [u'notamacropuseugenii', u'macropus_eugenii', u'macropus eugenii', u'tammar wallaby', u'neugenii', u'macropus', u'notamacropus eugenii', u'9315', u'macropuseugenii', u'noteug', u'meugenii', u'neug', u'meug', u'maceug', u'notamacropus', u'wallaby']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Tree Shrew', u'name': u'tupaia_belangeri', u'strain': None, u'common_name': u'northern tree shrew', u'accession': None, u'release': 96, u'groups': [u'core'], u'taxon_id': u'37347', u'assembly': u'TREESHREW', u'aliases': [u'tree_shrew', u'tupbel', u'37347', u'tupaia belangeri', u'tree shrew', u'northern tree shrew', u'tbelangeri', u'tbel']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'ASM359739v1', u'name': u'chelonoidis_abingdonii', u'common_name': u'Abingdon island giant tortoise', u'accession': u'GCA_003597395.1', u'release': 96, u'strain': None, u'groups': [u'core', u'rnaseq'], u'taxon_id': u'106734', u'display_name': u'Abingdon island giant tortoise', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Dord_2.0', u'name': u'dipodomys_ordii', u'common_name': u\"Ord's kangaroo rat\", u'accession': u'GCA_000151885.2', u'release': 96, u'strain': None, u'groups': [u'core', u'otherfeatures'], u'taxon_id': u'10020', u'display_name': u'Kangaroo rat', u'aliases': [u'dipord', u'10020', u'ords kangaroo rat', u'kangaroo rat', u'dordii', u'dipodomys ordii', u'dord', u\"ord's kangaroo rat\", u'kangaroo_rat', u'dipodomysordii']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Xenopus', u'name': u'xenopus_tropicalis', u'strain': None, u'accession': u'GCA_000004195.1', u'taxon_id': u'8364', u'common_name': u'western clawed frog', u'groups': [u'core', u'otherfeatures', u'funcgen'], u'release': 96, u'assembly': u'JGI_4.2', u'aliases': [u'xenopus', u'x.tropicalis', u'xentro', u'xenopus tropicalis', u'8364', u'xtro', u'xtropicalis', u'western clawed frog']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Red-bellied piranha', u'name': u'pygocentrus_nattereri', u'strain': None, u'accession': u'GCA_001682695.1', u'taxon_id': u'42514', u'release': 96, u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'common_name': u'red-bellied piranha', u'assembly': u'Pygocentrus_nattereri-1.0.2', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Tuatara', u'name': u'sphenodon_punctatus', u'strain': None, u'accession': u'GCA_003113815.1', u'taxon_id': u'8508', u'common_name': u'tuatara', u'groups': [u'rnaseq', u'core', u'otherfeatures'], u'release': 96, u'assembly': u'ASM311381v1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'C_variegatus-1.0', u'name': u'cyprinodon_variegatus', u'common_name': u'sheepshead minnow', u'accession': u'GCA_000732505.1', u'taxon_id': u'28743', u'strain': None, u'groups': [u'rnaseq', u'otherfeatures', u'core', u'funcgen'], u'release': 96, u'display_name': u'Sheepshead minnow', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'DMR_v1.0', u'name': u'fukomys_damarensis', u'accession': u'GCA_000743615.1', u'taxon_id': u'885580', u'strain': None, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'common_name': u'Damara mole rat', u'display_name': u'Damara mole rat', u'aliases': [u'fukomys_damarensi', u'fdam', u'fdamarensi', u'fukdam', u'damara mole rat', u'885580', u'fukomys damarensi'], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Bengalese finch', u'name': u'lonchura_striata_domestica', u'strain': None, u'accession': u'GCA_002197715.1', u'taxon_id': u'299123', u'release': 96, u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'common_name': u'Bengalese finch', u'assembly': u'LonStrDom1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'droNov1', u'name': u'dromaius_novaehollandiae', u'common_name': u'emu', u'accession': u'GCA_003342905.1', u'taxon_id': u'8790', u'strain': None, u'groups': [u'rnaseq', u'core', u'otherfeatures'], u'release': 96, u'display_name': u'Emu', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Tilapia', u'name': u'oreochromis_niloticus', u'strain': None, u'common_name': u'Nile tilapia', u'accession': u'GCA_000188235.1', u'release': 96, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'taxon_id': u'8128', u'assembly': u'Orenil1.0', u'aliases': [u'oniloticus', u'tilapia', u'orenil', u'oreochromis niloticus', u'8128', u'o. niloticus', u'nile tilapia', u'onil']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Prosim_1.0', u'name': u'prolemur_simus', u'common_name': u'greater bamboo lemur', u'accession': u'GCA_003258685.1', u'taxon_id': u'1328070', u'strain': None, u'groups': [u'core', u'funcgen'], u'release': 96, u'display_name': u'Greater bamboo lemur', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'assembly': u'C57BL_6NJ_v1', u'name': u'mus_musculus_c57bl6nj', u'common_name': u'house mouse', u'accession': u'GCA_001632555.1', u'taxon_id': u'10090', u'strain': u'C57BL/6NJ', u'groups': [u'core', u'funcgen'], u'release': 96, u'display_name': u'Mouse C57BL/6NJ', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'CSAV2.0', u'name': u'ciona_savignyi', u'taxon_id': u'51511', u'accession': None, u'release': 96, u'strain': None, u'groups': [u'otherfeatures', u'core'], u'common_name': u'Sea squirt Ciona savignyi', u'display_name': u'C.savignyi', u'aliases': [u'c.savignyi', u'ciona savignyi', u'csav', u'51511', u'sea squirt ciona savignyi', u'csavignyi', u'ciosav']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Japanese medaka HSOK', u'name': u'oryzias_latipes_hsok', u'strain': None, u'accession': u'GCA_002234695.1', u'common_name': u'Japanese medaka HSOK', u'assembly': u'ASM223469v1', u'groups': [u'rnaseq', u'core'], u'taxon_id': u'8090', u'release': 96, u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'assembly': u'CBA_J_v1', u'name': u'mus_musculus_cbaj', u'common_name': u'house mouse', u'accession': u'GCA_001624475.1', u'taxon_id': u'10090', u'strain': u'CBA/J', u'groups': [u'core', u'funcgen'], u'release': 96, u'display_name': u'Mouse CBA/J', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'MesAur1.0', u'name': u'mesocricetus_auratus', u'taxon_id': u'10036', u'accession': u'GCA_000349665.1', u'release': 96, u'strain': None, u'groups': [u'otherfeatures', u'core', u'funcgen'], u'common_name': u'Golden Hamster', u'display_name': u'Golden Hamster', u'aliases': [u'mesocricetus auratus', u'10036', u'mesaur', u'mauratus', u'maur', u'golden hamster']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Hyrax', u'name': u'procavia_capensis', u'strain': None, u'accession': None, u'taxon_id': u'9813', u'release': 96, u'groups': [u'core'], u'common_name': u'cape rock hyrax', u'assembly': u'proCap1', u'aliases': [u'procap', u'procavia capensis', u'rock_hyrax', u'9813', u'hyrax', u'pcap', u'procaviacapensis', u'cape rock hyrax', u'pcapensis']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Human', u'name': u'homo_sapiens', u'strain': None, u'accession': u'GCA_000001405.27', u'common_name': u'human', u'groups': [u'variation', u'cdna', u'rnaseq', u'funcgen', u'otherfeatures', u'core'], u'taxon_id': u'9606', u'assembly': u'GRCh38', u'aliases': [u'hsapiens', u'homsap', u'hsap', u'enshs', u'human', u'homo sapiens', u'h_sapiens', u'homo', u'9606'], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Capuchin', u'name': u'cebus_capucinus', u'strain': None, u'accession': u'GCA_001604975.1', u'taxon_id': u'1737458', u'common_name': u'White-headed capuchin', u'groups': [u'rnaseq', u'funcgen', u'core', u'otherfeatures'], u'release': 96, u'assembly': u'Cebus_imitator-1.0', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'aptRow1', u'name': u'apteryx_rowi', u'common_name': u'Okarito brown kiwi', u'accession': u'GCA_003343035.1', u'release': 96, u'strain': None, u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'taxon_id': u'308060', u'display_name': u'Okarito brown kiwi', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'CroPor_comp1', u'name': u'crocodylus_porosus', u'display_name': u'Australian saltwater crocodile', u'common_name': u'Australian saltwater crocodile', u'accession': u'GCA_001723895.1', u'strain': None, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'taxon_id': u'8502', u'release': 96, u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Coturnix_japonica_2.0', u'name': u'coturnix_japonica', u'common_name': u'Japanese quail', u'accession': u'GCA_001577835.1', u'release': 96, u'strain': None, u'groups': [u'core', u'otherfeatures', u'rnaseq'], u'taxon_id': u'93934', u'display_name': u'Japanese quail', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Mmur_3.0', u'name': u'microcebus_murinus', u'taxon_id': u'30608', u'accession': u'GCA_000165445.3', u'release': 96, u'strain': None, u'groups': [u'rnaseq', u'core', u'otherfeatures', u'funcgen'], u'common_name': u'gray mouse lemur', u'display_name': u'Mouse Lemur', u'aliases': [u'mmur', u'mmurinus', u'grey mouse lemur', u'mouse lemur', u'gray mouse lemur', u'micmur', u'mouse_lemur', u'microcebus murinus', u'30608']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Sheep', u'name': u'ovis_aries', u'strain': None, u'common_name': u'Domestic sheep', u'accession': u'GCA_000298735.1', u'release': 96, u'groups': [u'rnaseq', u'variation', u'otherfeatures', u'core'], u'taxon_id': u'9940', u'assembly': u'Oar_v3.1', u'aliases': [u'oviari', u'9940', u'sheep', u'domestic sheep', u'ovisaries', u'oari', u'oaries', u'oar', u'ovis aries']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Felis_catus_9.0', u'name': u'felis_catus', u'common_name': u'domestic cat', u'accession': u'GCA_000181335.4', u'strain': None, u'groups': [u'rnaseq', u'variation', u'core', u'otherfeatures', u'funcgen'], u'taxon_id': u'9685', u'release': 96, u'aliases': [u'fcat', u'cat', u'felcat', u'9685', u'domestic cat', u'felis catus', u'fcatus'], u'display_name': u'Cat'}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'SpeTri2.0', u'name': u'ictidomys_tridecemlineatus', u'accession': u'GCA_000236235.1', u'taxon_id': u'43179', u'strain': None, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'common_name': u'thirteen-lined ground squirrel', u'release': 96, u'aliases': [u'spermophilus_tridecemlineatus', u'spetri', u'squirrel', u'43179', u'thirteen-lined ground squirrel', u'stri', u'ictidomys tridecemlineatus', u'stridecemlineatus'], u'display_name': u'Squirrel'}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Nemo_v1', u'name': u'amphiprion_percula', u'taxon_id': u'161767', u'accession': u'GCA_003047355.1', u'release': 96, u'strain': None, u'groups': [u'core', u'rnaseq'], u'common_name': u'orange clownfish', u'display_name': u'Orange clownfish', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u\"Burton's mouthbrooder\", u'name': u'haplochromis_burtoni', u'strain': None, u'common_name': u\"Burton's mouthbrooder\", u'accession': u'GCA_000239415.1', u'release': 96, u'groups': [u'core', u'otherfeatures', u'rnaseq'], u'taxon_id': u'8153', u'assembly': u'AstBur1.0', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Goat', u'name': u'capra_hircus', u'strain': None, u'accession': u'GCA_001704415.1', u'taxon_id': u'9925', u'release': 96, u'groups': [u'otherfeatures', u'core', u'funcgen', u'rnaseq', u'variation'], u'common_name': u'Goat', u'assembly': u'ARS1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Cow', u'name': u'bos_taurus', u'strain': None, u'accession': u'GCA_002263795.2', u'taxon_id': u'9913', u'release': 96, u'groups': [u'variation', u'rnaseq', u'funcgen', u'otherfeatures', u'core'], u'common_name': u'cattle', u'assembly': u'ARS-UCD1.2', u'aliases': [u'cow', u'cattle', u'bos taurus', u'domestic cattle', u'9913', u'bovine', u'domestic cow', u'bostau', u'btau', u'btaurus']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'EquCab3.0', u'name': u'equus_caballus', u'common_name': u'horse', u'accession': u'GCA_002863925.1', u'release': 96, u'strain': None, u'groups': [u'core', u'otherfeatures', u'funcgen', u'rnaseq', u'variation'], u'taxon_id': u'9796', u'display_name': u'Horse', u'aliases': [u'equus caballus', u'ecab', u'9796', u'horse', u'equcab', u'equuscaballus', u'ecaballus']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Western mosquitofish', u'name': u'gambusia_affinis', u'strain': None, u'accession': u'GCA_003097735.1', u'taxon_id': u'33528', u'common_name': u'western mosquitofish', u'groups': [u'rnaseq', u'core'], u'release': 96, u'assembly': u'ASM309773v1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'WBcel235', u'name': u'caenorhabditis_elegans', u'common_name': u'C.elegans', u'accession': u'GCA_000002985.3', u'taxon_id': u'6239', u'strain': u'N2', u'groups': [u'core', u'funcgen'], u'release': 96, u'display_name': u'Caenorhabditis elegans', u'aliases': [u'caeele', u'caenorhabditis_elegans_prjna13758', u'cele', u'celegans', u'elegans', u'worm', u'6239', u'c.elegans', u'caenorhabditis elegans']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'TETRAODON8', u'name': u'tetraodon_nigroviridis', u'display_name': u'Tetraodon', u'accession': None, u'taxon_id': u'99883', u'strain': None, u'groups': [u'variation', u'core', u'otherfeatures'], u'common_name': u'spotted green pufferfish', u'release': 96, u'aliases': [u'tetnig', u'spotted green pufferfish', u'tnig', u'fresh water pufferfish', u'tetraodon', u'tetraodon nigroviridis', u'tnigroviridis', u'99883']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Degu', u'name': u'octodon_degus', u'strain': None, u'accession': u'GCA_000260255.1', u'taxon_id': u'10160', u'release': 96, u'groups': [u'core', u'otherfeatures'], u'common_name': u'Degu', u'assembly': u'OctDeg1.0', u'aliases': [u'10160', u'degu', u'octdeg', u'odegus', u'octodon degus', u'odeg']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Polar bear', u'name': u'ursus_maritimus', u'strain': None, u'common_name': u'Polar bear', u'accession': u'GCA_000687225.1', u'release': 96, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'taxon_id': u'29073', u'assembly': u'UrsMar_1.0', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'P_latipinna-1.0', u'name': u'poecilia_latipinna', u'common_name': u'sailfin molly', u'accession': u'GCA_001443285.1', u'release': 96, u'strain': None, u'groups': [u'core', u'otherfeatures', u'rnaseq'], u'taxon_id': u'48699', u'display_name': u'Sailfin molly', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Cang.pa_1.0', u'name': u'colobus_angolensis_palliatus', u'common_name': u'Angola colobus', u'accession': u'GCA_000951035.1', u'taxon_id': u'336983', u'strain': None, u'groups': [u'otherfeatures', u'core', u'funcgen'], u'release': 96, u'display_name': u'Angola colobus', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Lyretail cichlid', u'name': u'neolamprologus_brichardi', u'strain': None, u'accession': u'GCA_000239395.1', u'taxon_id': u'32507', u'release': 96, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'common_name': u'lyretail cichlid', u'assembly': u'NeoBri1.0', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'ailMel1', u'name': u'ailuropoda_melanoleuca', u'common_name': u'giant panda', u'accession': u'GCA_000004335.1', u'taxon_id': u'9646', u'strain': None, u'groups': [u'otherfeatures', u'core'], u'release': 96, u'display_name': u'Panda', u'aliases': [u'amelanoleuca', u'ailuropoda melanoleuca', u'ailmel1', u'amel', u'panda', u'9646', u'ailmel', u'giant panda']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Mnem_1.0', u'name': u'macaca_nemestrina', u'common_name': u'Pig-tailed macaque', u'accession': u'GCA_000956065.1', u'taxon_id': u'9545', u'strain': None, u'groups': [u'rnaseq', u'funcgen', u'otherfeatures', u'core'], u'release': 96, u'display_name': u'Pig-tailed macaque', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'LatCha1', u'name': u'latimeria_chalumnae', u'taxon_id': u'7897', u'accession': u'GCA_000225785.1', u'release': 96, u'strain': None, u'groups': [u'otherfeatures', u'core'], u'common_name': u'coelacanth', u'display_name': u'Coelacanth', u'aliases': [u'lcha', u'lchalumnae', u'coelacanth', u'lamineria', u'latimeria chalumnae', u'7897', u'latcha', u'living fossil']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Duck', u'name': u'anas_platyrhynchos_platyrhynchos', u'strain': None, u'accession': u'GCA_002743455.1', u'taxon_id': u'8840', u'release': 96, u'groups': [u'core', u'funcgen', u'rnaseq'], u'common_name': u'common mallard', u'assembly': u'CAU_duck1.0', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Donkey', u'name': u'equus_asinus_asinus', u'strain': None, u'accession': u'GCA_003033725.1', u'taxon_id': u'83772', u'release': 96, u'groups': [u'core', u'rnaseq'], u'common_name': u'donkey', u'assembly': u'ASM303372v1', u'aliases': [u'donkey', u'ass']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Climbing perch', u'name': u'anabas_testudineus', u'strain': None, u'accession': u'GCA_900324465.1', u'common_name': u'climbing perch', u'groups': [u'core', u'rnaseq'], u'taxon_id': u'64144', u'assembly': u'fAnaTes1.1', u'aliases': [], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Blue tit', u'name': u'cyanistes_caeruleus', u'strain': None, u'accession': u'GCA_002901205.1', u'common_name': u'blue tit', u'groups': [u'rnaseq', u'core', u'otherfeatures'], u'taxon_id': u'156563', u'assembly': u'cyaCae2', u'aliases': [], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Rnor_6.0', u'name': u'rattus_norvegicus', u'common_name': u'Norway rat', u'accession': u'GCA_000001895.4', u'release': 96, u'strain': None, u'groups': [u'rnaseq', u'variation', u'core', u'otherfeatures', u'funcgen'], u'taxon_id': u'10116', u'display_name': u'Rat', u'aliases': [u'rattus', u'rnor', u'rattus norvegicus', u'r_norvegicus', u'rnorvegicus', u'norway rat', u'ratnor', u'rat', u'10116']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Cavpor3.0', u'name': u'cavia_porcellus', u'common_name': u'domestic guinea pig', u'accession': u'GCA_000151735.1', u'taxon_id': u'10141', u'strain': None, u'groups': [u'rnaseq', u'funcgen', u'otherfeatures', u'core'], u'release': 96, u'display_name': u'Guinea Pig', u'aliases': [u'cavia porcellus', u'guinea pig', u'cpor', u'guinea_pig', u'cporcellus', u'cavpor', u'10141', u'domestic guinea pig']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'IpCoco_1.2', u'name': u'ictalurus_punctatus', u'common_name': u'channel catfish', u'accession': u'GCA_001660625.1', u'strain': None, u'groups': [u'funcgen', u'otherfeatures', u'core', u'rnaseq'], u'taxon_id': u'7998', u'release': 96, u'aliases': [], u'display_name': u'Channel catfish'}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'ASM171598v2', u'name': u'manacus_vitellinus', u'common_name': u'golden-collared manakin', u'accession': u'GCA_001715985.2', u'strain': None, u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'taxon_id': u'328815', u'display_name': u'Golden-collared manakin', u'aliases': [], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'R64-1-1', u'name': u'saccharomyces_cerevisiae', u'display_name': u'Saccharomyces cerevisiae', u'common_name': u\"baker's yeast\", u'accession': u'GCA_000146045.2', u'strain': u'S288C', u'groups': [u'funcgen', u'core', u'otherfeatures', u'variation'], u'taxon_id': u'4932', u'release': 96, u'aliases': [u\"saccharomyces cerevisiae (baker's yeast)\", u'4932', u'saccharomyces cerevisiae s288c', u\"baker's yeast\", u'saccharomyces cerevisiae', u'scerevisiae', u'saccer', u's_cerevisiae', u'scer']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Platypus', u'name': u'ornithorhynchus_anatinus', u'strain': None, u'accession': u'GCF_000002275.2', u'taxon_id': u'9258', u'common_name': u'platypus', u'groups': [u'rnaseq', u'variation', u'core', u'otherfeatures', u'funcgen'], u'release': 96, u'assembly': u'OANA5', u'aliases': [u'platypus', u'oana', u'ornithorhynchus anatinus', u'9258', u'ornana', u'oanatinus']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Eastern happy', u'name': u'astatotilapia_calliptera', u'strain': None, u'accession': u'GCA_900246225.3', u'taxon_id': u'8154', u'release': 96, u'groups': [u'core', u'rnaseq'], u'common_name': u'eastern happy', u'assembly': u'fAstCal1.2', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'BosGru_v2.0', u'name': u'bos_mutus', u'display_name': u'Wild yak', u'common_name': u'wild yak', u'accession': u'GCA_000298355.1', u'strain': None, u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'taxon_id': u'72004', u'release': 96, u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Black snub-nosed monkey', u'name': u'rhinopithecus_bieti', u'strain': None, u'accession': u'GCA_001698545.1', u'common_name': u'Black snub-nosed monkey', u'assembly': u'ASM169854v1', u'groups': [u'rnaseq', u'otherfeatures', u'core', u'funcgen'], u'taxon_id': u'61621', u'release': 96, u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Fugu', u'name': u'takifugu_rubripes', u'strain': None, u'accession': u'GCA_000180615.2', u'taxon_id': u'31033', u'release': 96, u'groups': [u'rnaseq', u'core', u'otherfeatures'], u'common_name': u'torafugu', u'assembly': u'FUGU5', u'aliases': [u'trub', u'takifugu', u't_rubripes', u'trubripes', u'takifugu rubripes', u'takrub', u'31033', u'torafugu', u'fugu']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Red fox', u'name': u'vulpes_vulpes', u'strain': None, u'common_name': u'red fox', u'accession': u'GCA_003160815.1', u'release': 96, u'groups': [u'core', u'otherfeatures', u'rnaseq'], u'taxon_id': u'9627', u'assembly': u'VulVul2.2', u'aliases': [u'silver fox', u'fox']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'gorGor4', u'name': u'gorilla_gorilla', u'display_name': u'Gorilla', u'common_name': u'Western Lowland Gorilla', u'accession': u'GCA_000151905.3', u'strain': None, u'groups': [u'funcgen', u'core', u'otherfeatures', u'rnaseq'], u'taxon_id': u'9595', u'release': 96, u'aliases': [u'western gorilla', u'gorilla', u'gorilla gorilla gorilla', u'ggor', u'9595', u'ggorilla', u'9593', u'gorilla_gorilla_gorilla', u'gorgor', u'gorilla gorilla']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'gadMor1', u'name': u'gadus_morhua', u'common_name': u'Atlantic cod', u'accession': None, u'taxon_id': u'8049', u'strain': None, u'groups': [u'core'], u'release': 96, u'display_name': u'Cod', u'aliases': [u'atlantic cod', u'8049', u'gadmor', u'cod', u'atlantic_cod', u'gmor', u'gmorhua', u'gadus morhua']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Central bearded dragon', u'name': u'pogona_vitticeps', u'strain': None, u'accession': u'GCA_900067755.1', u'taxon_id': u'103695', u'release': 96, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'common_name': u'central bearded dragon', u'assembly': u'pvi1.1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Megabat', u'name': u'pteropus_vampyrus', u'strain': None, u'accession': None, u'taxon_id': u'132908', u'common_name': u'large flying fox', u'groups': [u'core'], u'release': 96, u'assembly': u'pteVam1', u'aliases': [u'pvam', u'pvampyrus', u'ptevam', u'megabat', u'large flying fox', u'pteropus vampyrus', u'pteropusvampyrus', u'132908']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'American bison', u'name': u'bison_bison_bison', u'strain': None, u'common_name': u'American bison', u'accession': u'GCA_000754665.1', u'release': 96, u'groups': [u'core', u'otherfeatures'], u'taxon_id': u'43346', u'assembly': u'Bison_UMD1.0', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Tarsius_syrichta-2.0.1', u'name': u'carlito_syrichta', u'display_name': u'Tarsier', u'accession': u'GCA_000164805.2', u'taxon_id': u'1868482', u'strain': None, u'groups': [u'funcgen', u'otherfeatures', u'core'], u'common_name': u'Philippine tarsier', u'release': 96, u'aliases': [u'carsyr', u'tsyr', u'philippine tarsier', u'tarsius syrichta', u'csyrichta', u'tsyrichta', u'carlito', u'csyr', u'tarsier', u'tarsius_syrichta', u'carlitosyrichta', u'tarsiussyrichta', u'9478', u'tarsyr', u'carlito syrichta']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'PM.fa', u'name': u'periophthalmus_magnuspinnatus', u'common_name': u'Periophthalmus magnuspinnatus', u'accession': u'GCA_000787105.1', u'taxon_id': u'409849', u'strain': None, u'groups': [u'core'], u'release': 96, u'display_name': u'Periophthalmus magnuspinnatus', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'ASM259213v1', u'name': u'anser_brachyrhynchus', u'accession': u'GCA_002592135.1', u'taxon_id': u'132585', u'strain': None, u'groups': [u'core'], u'common_name': u'pink-footed goose', u'display_name': u'Pink-footed goose', u'aliases': [], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'CriGri-PICR', u'name': u'cricetulus_griseus_picr', u'common_name': u'Chinese hamster', u'accession': u'GCA_003668045.1', u'release': 96, u'strain': None, u'groups': [u'rnaseq', u'funcgen', u'core'], u'taxon_id': u'10029', u'display_name': u'Chinese hamster PICR', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Zig-zag eel', u'name': u'mastacembelus_armatus', u'strain': None, u'accession': u'GCA_900324485.1', u'common_name': u'zig-zag eel', u'assembly': u'fMasArm1.1', u'groups': [u'rnaseq', u'core'], u'taxon_id': u'205130', u'release': 96, u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'BallGen_V1', u'name': u'labrus_bergylta', u'common_name': u'ballan wrasse', u'accession': u'GCA_900080235.1', u'release': 96, u'strain': None, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'taxon_id': u'56723', u'display_name': u'Ballan wrasse', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Naked mole-rat female', u'name': u'heterocephalus_glaber_female', u'strain': None, u'common_name': u'naked mole-rat', u'accession': u'GCA_000247695.1', u'release': 96, u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'taxon_id': u'10181', u'assembly': u'HetGla_female_1.0', u'aliases': [u'naked mole-rat female', u'hglaber_female', u'heterocephalus glaber female']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Shortfin molly', u'name': u'poecilia_mexicana', u'strain': None, u'accession': u'GCA_001443325.1', u'common_name': u'shortfin molly', u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'taxon_id': u'48701', u'assembly': u'P_mexicana-1.0', u'aliases': [], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'ASM223471v1', u'name': u'oryzias_latipes_hni', u'display_name': u'Japanese medaka HNI', u'common_name': u'Japanese medaka HNI', u'accession': u'GCA_002234715.1', u'strain': None, u'groups': [u'core', u'rnaseq'], u'taxon_id': u'8090', u'release': 96, u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Indian medaka', u'name': u'oryzias_melastigma', u'strain': None, u'common_name': u'Indian medaka', u'accession': u'GCA_002922805.1', u'release': 96, u'groups': [u'rnaseq', u'core'], u'taxon_id': u'30732', u'assembly': u'Om_v0.7.RACA', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'assembly': u'A_J_v1', u'name': u'mus_musculus_aj', u'common_name': u'house mouse', u'accession': u'GCA_001624215.1', u'taxon_id': u'10090', u'strain': u'A/J', u'groups': [u'core', u'funcgen'], u'release': 96, u'display_name': u'Mouse A/J', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Zonotrichia_albicollis-1.0.1', u'name': u'zonotrichia_albicollis', u'accession': u'GCA_000385455.1', u'taxon_id': u'44394', u'strain': None, u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'common_name': u'white-throated sparrow', u'display_name': u'White-throated sparrow', u'aliases': [], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Bonobo', u'name': u'pan_paniscus', u'strain': None, u'accession': u'GCA_000258655.2', u'common_name': u'bonobo', u'groups': [u'funcgen', u'core', u'otherfeatures', u'rnaseq'], u'taxon_id': u'9597', u'assembly': u'panpan1.1', u'aliases': [], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'OryCun2.0', u'name': u'oryctolagus_cuniculus', u'common_name': u'rabbit', u'accession': u'GCA_000003625.1', u'release': 96, u'strain': None, u'groups': [u'rnaseq', u'otherfeatures', u'core', u'funcgen'], u'taxon_id': u'9986', u'display_name': u'Rabbit', u'aliases': [u'ocuniculus', u'9986', u'ocun', u'orycun', u'oryctolagus cuniculus', u'rabbit']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'GRCg6a', u'name': u'gallus_gallus', u'accession': u'GCA_000002315.5', u'taxon_id': u'9031', u'strain': None, u'groups': [u'funcgen', u'core', u'otherfeatures', u'variation', u'rnaseq'], u'common_name': u'chicken', u'release': 96, u'aliases': [u'g_gallus', u'chicken', u'galgal', u'9031', u'ggallus', u'ggal', u'gallus gallus'], u'display_name': u'Chicken'}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'ASM223467v1', u'name': u'oryzias_latipes', u'display_name': u'Japanese medaka HdrR', u'accession': u'GCA_002234675.1', u'taxon_id': u'8090', u'strain': None, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'common_name': u'Japanese medaka HdrR', u'release': 96, u'aliases': [u'oryzias latipes', u'8090', u'olat', u'olatipes', u'japanese medaka', u'medaka', u'orylat']}, {u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'assembly': u'WSB_EiJ_v1', u'name': u'mus_musculus_wsbeij', u'common_name': u'western european house mouse', u'accession': u'GCA_001624835.1', u'taxon_id': u'10092', u'strain': u'WSB/EiJ', u'groups': [u'funcgen', u'core'], u'release': 96, u'display_name': u'Mouse WSB/EiJ', u'aliases': [u'mus_musculus_domesticus_wsbeij', u'10092', u'western european house mouse', u'mus musculus domesticus wsbeij', u'mmusculus_domesticus_wsbeij', u'mouse wsbeij']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Zebra mbuna', u'name': u'maylandia_zebra', u'strain': None, u'accession': u'GCA_000238955.5', u'taxon_id': u'106582', u'common_name': u'zebra mbuna', u'groups': [u'rnaseq', u'core', u'otherfeatures'], u'release': 96, u'assembly': u'M_zebra_UMD2a', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'ASM164957v1', u'name': u'kryptolebias_marmoratus', u'display_name': u'Mangrove rivulus', u'accession': u'GCA_001649575.1', u'taxon_id': u'37003', u'strain': None, u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'common_name': u'mangrove rivulus', u'release': 96, u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Turbot', u'name': u'scophthalmus_maximus', u'strain': None, u'common_name': u'turbot', u'accession': u'GCA_003186165.1', u'release': 96, u'groups': [u'rnaseq', u'funcgen', u'core'], u'taxon_id': u'52904', u'assembly': u'ASM318616v1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u\"Coquerel's sifaka\", u'name': u'propithecus_coquereli', u'strain': None, u'common_name': u\"Coquerel's sifaka\", u'accession': u'GCA_000956105.1', u'release': 96, u'groups': [u'core', u'otherfeatures', u'funcgen'], u'taxon_id': u'379532', u'assembly': u'Pcoq_1.0', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Clown anemonefish', u'name': u'amphiprion_ocellaris', u'strain': None, u'accession': u'GCA_002776465.1', u'taxon_id': u'80972', u'release': 96, u'groups': [u'rnaseq', u'core', u'otherfeatures'], u'common_name': u'clown anemonefish', u'assembly': u'AmpOce1.0', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'BDGP6.22', u'name': u'drosophila_melanogaster', u'common_name': u'fruit fly', u'accession': u'GCA_000001215.4', u'strain': None, u'groups': [u'funcgen', u'core'], u'taxon_id': u'7227', u'release': 96, u'aliases': [u'd.mel', u'dmel', u'd. melanogaster'], u'display_name': u'Drosophila melanogaster'}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Upper Galilee mountains blind mole rat', u'name': u'nannospalax_galili', u'strain': None, u'accession': u'GCA_000622305.1', u'common_name': u'Upper Galilee mountains blind mole rat', u'assembly': u'S.galili_v1.0', u'groups': [u'rnaseq', u'funcgen', u'otherfeatures', u'core'], u'taxon_id': u'1026970', u'release': 96, u'aliases': [u'ngalili', u'nangal', u'upper galilee mountains blind mole rat', u'1026970', u'nannospalax galili', u'ngal']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Stegastes_partitus-1.0.2', u'name': u'stegastes_partitus', u'common_name': u'bicolor damselfish', u'accession': u'GCA_000690725.1', u'strain': None, u'groups': [u'core', u'otherfeatures', u'rnaseq'], u'taxon_id': u'144197', u'display_name': u'Bicolor damselfish', u'aliases': [], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Spiny chromis', u'name': u'acanthochromis_polyacanthus', u'strain': None, u'accession': u'GCA_002109545.1', u'common_name': u'spiny chromis', u'assembly': u'ASM210954v1', u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'taxon_id': u'80966', u'release': 96, u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Asian bonytongue', u'name': u'scleropages_formosus', u'strain': None, u'accession': u'GCA_001624265.1', u'taxon_id': u'113540', u'common_name': u'Asian bonytongue', u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'release': 96, u'assembly': u'ASM162426v1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'notPer1', u'name': u'nothoprocta_perdicaria', u'common_name': u'birds', u'accession': u'GCA_003342845.1', u'strain': None, u'groups': [u'core', u'otherfeatures'], u'taxon_id': u'30464', u'display_name': u'Chilean tinamou', u'aliases': [], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'assembly': u'CAST_EiJ_v1', u'name': u'mus_musculus_casteij', u'common_name': u'south eastern house mouse', u'accession': u'GCA_001624445.1', u'release': 96, u'strain': u'CAST/EiJ', u'groups': [u'funcgen', u'core'], u'taxon_id': u'10091', u'display_name': u'Mouse CAST/EiJ', u'aliases': [u'mus_musculus_castaneus_casteij', u'south eastern house mouse', u'10091', u'mmusculus_castaneus_casteij', u'mus musculus castaneus casteij', u'mouse casteij']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Tiger tail seahorse', u'name': u'hippocampus_comes', u'strain': None, u'accession': u'GCA_001891065.1', u'taxon_id': u'109280', u'release': 96, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'common_name': u'tiger tail seahorse', u'assembly': u'H_comes_QL1_v1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'CavAp1.0', u'name': u'cavia_aperea', u'common_name': u'Brazilian guinea pig', u'accession': u'GCA_000688575.1', u'taxon_id': u'37548', u'strain': None, u'groups': [u'core'], u'release': 96, u'display_name': u'Brazilian guinea pig', u'aliases': [u'brazilian guinea pig', u'cape', u'cavape', u'cavia aperea', u'caperea', u'37548']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'turTru1', u'name': u'tursiops_truncatus', u'common_name': u'bottlenosed dolphin', u'accession': None, u'strain': None, u'groups': [u'core'], u'taxon_id': u'9739', u'release': 96, u'aliases': [u'tursiops truncatus', u'9739', u'turtru', u'bottlenose dolphin', u'bottlenosed dolphin', u'dolphin', u'ttruncatus', u'tursiopstruncatus', u'ttru'], u'display_name': u'Dolphin'}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'taeGut3.2.4', u'name': u'taeniopygia_guttata', u'display_name': u'Zebra Finch', u'accession': None, u'taxon_id': u'59729', u'strain': None, u'groups': [u'variation', u'otherfeatures', u'core'], u'common_name': u'zebra finch', u'release': 96, u'aliases': [u'zebra finch', u'taeniopygia guttata', u'zebrafinch', u'finch', u'taeniopygiaguttata', u'tguttata', u'taeniopygia', u'59729', u'tgut', u'taegut']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Gelada', u'name': u'theropithecus_gelada', u'strain': None, u'accession': u'GCA_003255815.1', u'common_name': u'gelada', u'assembly': u'Tgel_1.0', u'groups': [u'rnaseq', u'core', u'funcgen'], u'taxon_id': u'9565', u'release': 96, u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Flycatcher', u'name': u'ficedula_albicollis', u'strain': None, u'common_name': u'Collared flycatcher', u'accession': u'GCA_000247815.1', u'release': 96, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'taxon_id': u'59894', u'assembly': u'FicAlb_1.4', u'aliases': [u'f_albicollis', u'ficedula albicollis', u'collared flycatcher', u'flycatcher', u'59894', u'ficalb', u'falbicollis', u'falb']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Vervet-AGM', u'name': u'chlorocebus_sabaeus', u'strain': None, u'common_name': u'African green monkey', u'accession': u'GCA_000409795.2', u'release': 96, u'groups': [u'core', u'otherfeatures', u'rnaseq', u'variation'], u'taxon_id': u'60711', u'assembly': u'ChlSab1.1', u'aliases': [u'agm', u'vervet monkey', u'60711', u'csabaeus', u'chlorocebus sabeus', u'csab', u'chlsab', u'chlorocebus_sabeus', u'vervet-agm', u'african green monkey', u'chlorocebus sabaeus']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'monDom5', u'name': u'monodelphis_domestica', u'common_name': u'gray short-tailed opossum', u'accession': u'GCF_000002295.2', u'taxon_id': u'13616', u'strain': None, u'groups': [u'core', u'variation', u'rnaseq'], u'release': 96, u'display_name': u'Opossum', u'aliases': [u'mondom', u'gray short-tailed opossum', u'broado5', u'opossum', u'mdomestica', u'13616', u'monodelphis domestica', u'mdom']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'BROADS1', u'name': u'gasterosteus_aculeatus', u'common_name': u'three-spined stickleback', u'accession': None, u'release': 96, u'strain': None, u'groups': [u'core', u'otherfeatures'], u'taxon_id': u'69293', u'display_name': u'Stickleback', u'aliases': [u'three-spined stickleback', u'gacu', u'gaculeatus', u'gasacu', u'gasterosteus aculeatus', u'three spined stickleback', u'stickleback', u'69293']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Tiger', u'name': u'panthera_tigris_altaica', u'strain': None, u'common_name': u'Tiger', u'accession': u'GCA_000464555.1', u'release': 96, u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'taxon_id': u'74533', u'assembly': u'PanTig1.0', u'aliases': [u'tiger']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Midas_v5', u'name': u'amphilophus_citrinellus', u'common_name': u'Midas cichlid', u'accession': u'GCA_000751415.1', u'strain': None, u'groups': [u'core'], u'taxon_id': u'61819', u'release': 96, u'aliases': [], u'display_name': u'Midas cichlid'}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Ugandan red Colobus', u'name': u'piliocolobus_tephrosceles', u'strain': None, u'accession': u'GCA_002776525.2', u'taxon_id': u'591936', u'release': 96, u'groups': [u'rnaseq', u'core', u'funcgen'], u'common_name': u'Ugandan red Colobus', u'assembly': u'ASM277652v2', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Chinese hamster CriGri', u'name': u'cricetulus_griseus_crigri', u'strain': None, u'common_name': u'Chinese hamster', u'accession': u'GCA_000223135.1', u'release': 96, u'groups': [u'core', u'otherfeatures', u'funcgen', u'rnaseq'], u'taxon_id': u'10029', u'assembly': u'CriGri_1.0', u'aliases': [u'cricetulus griseus crigri', u'cgriseus_crigri', u'chinese hamster crigri']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'CanFam3.1', u'name': u'canis_familiaris', u'common_name': u'dog', u'accession': u'GCA_000002285.2', u'taxon_id': u'9615', u'strain': None, u'groups': [u'rnaseq', u'variation', u'otherfeatures', u'core', u'funcgen'], u'release': 96, u'display_name': u'Dog', u'aliases': [u'dog', u'canis', u'cfam', u'c_familiaris', u'canfam', u'canis lupus familiaris', u'canis familiaris', u'9615', u'canis_lupus_familiaris', u'cfamiliaris']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'HEDGEHOG', u'name': u'erinaceus_europaeus', u'common_name': u'western European hedgehog', u'accession': None, u'strain': None, u'groups': [u'core'], u'taxon_id': u'9365', u'display_name': u'Hedgehog', u'aliases': [u'erinaceus europaeus', u'eeuropaeus', u'erieur', u'eeur', u'western european hedgehog', u'9365', u'hedgehog', u'western_european_hedgehog'], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'ASM382977v1', u'name': u'junco_hyemalis', u'taxon_id': u'40217', u'accession': u'GCA_003829775.1', u'release': 96, u'strain': None, u'groups': [u'core', u'rnaseq'], u'common_name': u'dark-eyed junco', u'display_name': u'Dark-eyed junco', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Ruff', u'name': u'calidris_pugnax', u'strain': None, u'accession': u'GCA_001431845.1', u'taxon_id': u'198806', u'common_name': u'ruff', u'groups': [u'core', u'otherfeatures', u'rnaseq'], u'release': 96, u'assembly': u'ASM143184v1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Dasnov3.0', u'name': u'dasypus_novemcinctus', u'common_name': u'nine-banded armadillo', u'accession': u'GCA_000208655.2', u'strain': None, u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'taxon_id': u'9361', u'display_name': u'Armadillo', u'aliases': [u'dasnov', u'dnov', u'9361', u'arma', u'dasypus novemcinctu', u'nine-banded armadillo', u'dasypus novemcinctus', u'armadillo', u'dnovemcinctus'], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'PAHARI_EIJ_v1.1', u'name': u'mus_pahari', u'common_name': u'Shrew mouse', u'accession': u'GCA_900095145.2', u'release': 96, u'strain': u'PAHARI_EIJ', u'groups': [u'core'], u'taxon_id': u'10093', u'display_name': u'Shrew mouse', u'aliases': [u'shrew mouse', u'mpah', u'mpahari', u'muspah', u'mus pahari', u'10093']}, {u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'display_name': u'Mouse 129S1/SvImJ', u'name': u'mus_musculus_129s1svimj', u'strain': u'129S1/SvImJ', u'common_name': u'house mouse', u'accession': u'GCA_001624185.1', u'release': 96, u'groups': [u'funcgen', u'core'], u'taxon_id': u'10090', u'assembly': u'129S1_SvImJ_v1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'assembly': u'AKR_J_v1', u'name': u'mus_musculus_akrj', u'common_name': u'house mouse', u'accession': u'GCA_001624295.1', u'strain': u'AKR/J', u'groups': [u'funcgen', u'core'], u'taxon_id': u'10090', u'release': 96, u'aliases': [], u'display_name': u'Mouse AKR/J'}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Macaque', u'name': u'macaca_mulatta', u'strain': None, u'accession': u'GCA_000772875.3', u'common_name': u'rhesus monkey', u'assembly': u'Mmul_8.0.1', u'groups': [u'funcgen', u'otherfeatures', u'core', u'variation', u'rnaseq'], u'taxon_id': u'9544', u'release': 96, u'aliases': [u'macaque', u'rhesus_monkey', u'9544', u'rhesus monkey', u'mmul', u'rhesus', u'macaca mulatta', u'macmul', u'mmulatta']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Bushbaby', u'name': u'otolemur_garnettii', u'strain': None, u'accession': u'GCA_000181295.3', u'common_name': u'small-eared galago', u'groups': [u'core', u'otherfeatures'], u'taxon_id': u'30611', u'assembly': u'OtoGar3', u'aliases': [u'galago', u'otolemur garnettii', u'otogar', u'ogarnettii', u'ogar', u'small-eared galago', u'bushbaby', u'30611'], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Naked mole-rat male', u'name': u'heterocephalus_glaber_male', u'strain': None, u'accession': u'GCA_000230445.1', u'taxon_id': u'10181', u'release': 96, u'groups': [u'core', u'otherfeatures', u'rnaseq'], u'common_name': u'naked mole-rat', u'assembly': u'HetGla_1.0', u'aliases': [u'hgla', u'naked mole-rat', u'hetgla', u'10181', u'hglaber_male', u'heterocephalus glaber male', u'naked mole-rat male']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'C.can_genome_v1.0', u'name': u'castor_canadensis', u'common_name': u'American beaver', u'accession': u'GCA_001984765.1', u'strain': None, u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'taxon_id': u'51338', u'release': 96, u'aliases': [], u'display_name': u'American beaver'}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'PKINGS_0.1', u'name': u'paramormyrops_kingsleyae', u'common_name': u'Paramormyrops kingsleyae', u'accession': u'GCA_002872115.1', u'strain': None, u'groups': [u'rnaseq', u'core'], u'taxon_id': u'1676925', u'display_name': u'Paramormyrops kingsleyae', u'aliases': [], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Mainland tiger snake', u'name': u'notechis_scutatus', u'strain': None, u'accession': u'GCA_900518725.1', u'taxon_id': u'8663', u'common_name': u'mainland tiger snake', u'groups': [u'core', u'otherfeatures', u'rnaseq'], u'release': 96, u'assembly': u'TS10Xv2-PRI', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Daurian ground squirrel', u'name': u'spermophilus_dauricus', u'strain': None, u'accession': u'GCA_002406435.1', u'taxon_id': u'99837', u'common_name': u'Daurian ground squirrel', u'groups': [u'core'], u'release': 96, u'assembly': u'ASM240643v1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Algerian mouse', u'name': u'mus_spretus', u'strain': u'SPRET/EiJ', u'accession': u'GCA_001624865.1', u'common_name': u'algerian mouse', u'assembly': u'SPRET_EiJ_v1', u'groups': [u'core', u'funcgen'], u'taxon_id': u'10096', u'release': 96, u'aliases': [u'spreteij', u'mspretus', u'mus_spretus_spret_eij', u'western mediterranean mouse']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Guppy', u'name': u'poecilia_reticulata', u'strain': None, u'accession': u'GCA_000633615.2', u'taxon_id': u'8081', u'release': 96, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'common_name': u'guppy', u'assembly': u'Guppy_female_1.0_MT', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'MunDraft-v1.0', u'name': u'meriones_unguiculatus', u'common_name': u'Mongolian gerbil', u'accession': u'GCA_002204375.1', u'strain': None, u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'taxon_id': u'10047', u'release': 96, u'aliases': [], u'display_name': u'Mongolian gerbil'}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Blue-crowned manakin', u'name': u'lepidothrix_coronata', u'strain': None, u'accession': u'GCA_001604755.1', u'taxon_id': u'321398', u'release': 96, u'groups': [u'rnaseq', u'core', u'otherfeatures'], u'common_name': u'blue-crowned manakin', u'assembly': u'Lepidothrix_coronata-1.0', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Common canary', u'name': u'serinus_canaria', u'strain': None, u'common_name': u'common canary', u'accession': u'GCA_000534875.1', u'release': 96, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'taxon_id': u'9135', u'assembly': u'SCA1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Long-tailed chinchilla', u'name': u'chinchilla_lanigera', u'strain': None, u'accession': u'GCA_000276665.1', u'common_name': u'Long-tailed chinchilla', u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'taxon_id': u'34839', u'assembly': u'ChiLan1.0', u'aliases': [u'clan', u'34839', u'clanigera', u'chinchilla lanigera', u'chilan', u'long-tailed chinchilla'], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'AnoCar2.0', u'name': u'anolis_carolinensis', u'common_name': u'green anole', u'accession': u'GCA_000090745.1', u'strain': None, u'groups': [u'rnaseq', u'core', u'otherfeatures'], u'taxon_id': u'28377', u'display_name': u'Anole lizard', u'aliases': [u'acarolinensis', u'green anole lizard', u'anolis carolinensis', u'28377', u'anolis', u'green anole', u'acar', u'anole lizard', u'anocar', u'anolis_lizard', u'anoliscarolinensis'], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Turkey_2.01', u'name': u'meleagris_gallopavo', u'common_name': u'domestic turkey', u'accession': u'GCA_000146605.1', u'taxon_id': u'9103', u'strain': None, u'groups': [u'core', u'otherfeatures', u'variation'], u'release': 96, u'display_name': u'Turkey', u'aliases': [u'wild turkey', u'melgal', u'mgal', u'mgallopavo', u'9103', u'turkey', u'meleagris gallopavo', u'mga', u'common turkey']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'phaCin_unsw_v4.1', u'name': u'phascolarctos_cinereus', u'common_name': u'koala', u'accession': u'GCA_002099425.1', u'strain': None, u'groups': [u'core'], u'taxon_id': u'38626', u'display_name': u'Koala', u'aliases': [], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'NumMel1.0', u'name': u'numida_meleagris', u'display_name': u'Helmeted guineafowl', u'accession': u'GCA_002078875.2', u'taxon_id': u'8996', u'strain': None, u'groups': [u'rnaseq', u'core', u'otherfeatures'], u'common_name': u'helmeted guineafowl', u'release': 96, u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Sscrofa11.1', u'name': u'sus_scrofa', u'accession': u'GCA_000003025.6', u'taxon_id': u'9823', u'strain': None, u'groups': [u'funcgen', u'core', u'otherfeatures', u'variation', u'rnaseq'], u'common_name': u'pig', u'display_name': u'Pig', u'aliases': [u'pig', u'sscr', u'9823', u'sscrofa', u'swine', u'sus scrofa', u'wild boar', u'pigs'], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u\"Agassiz's desert tortoise\", u'name': u'gopherus_agassizii', u'strain': None, u'accession': u'GCA_002896415.1', u'taxon_id': u'38772', u'release': 96, u'groups': [u'rnaseq', u'core'], u'common_name': u\"Agassiz's desert tortoise\", u'assembly': u'ASM289641v1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'ASM325472v1', u'name': u'canis_lupus_dingo', u'common_name': u'dingo', u'accession': u'GCA_003254725.1', u'taxon_id': u'286419', u'strain': None, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'release': 96, u'display_name': u'Dingo', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Anan_2.0', u'name': u'aotus_nancymaae', u'common_name': u\"Ma's night monkey\", u'accession': u'GCA_000952055.2', u'taxon_id': u'37293', u'strain': None, u'groups': [u'rnaseq', u'otherfeatures', u'core', u'funcgen'], u'release': 96, u'display_name': u\"Ma's night monkey\", u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'MUSP714', u'name': u'mus_spicilegus', u'taxon_id': u'10103', u'accession': u'GCA_003336285.1', u'release': 96, u'strain': None, u'groups': [u'core', u'rnaseq'], u'common_name': u'steppe mouse', u'display_name': u'Steppe mouse', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Greater amberjack', u'name': u'seriola_dumerili', u'strain': None, u'accession': u'GCA_002260705.1', u'common_name': u'greater amberjack', u'assembly': u'Sdu_1.0', u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'taxon_id': u'41447', u'release': 96, u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Golden snub-nosed monkey', u'name': u'rhinopithecus_roxellana', u'strain': None, u'accession': u'GCA_000769185.1', u'taxon_id': u'61622', u'common_name': u'Golden snub-nosed monkey', u'groups': [u'core', u'otherfeatures', u'funcgen', u'rnaseq'], u'release': 96, u'assembly': u'Rrox_v1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'MicOch1.0', u'name': u'microtus_ochrogaster', u'common_name': u'vole', u'accession': u'GCA_000317375.1', u'strain': None, u'groups': [u'core', u'otherfeatures'], u'taxon_id': u'79684', u'release': 96, u'aliases': [u'prairie vole', u'vole', u'micoch', u'mochrogaster', u'microtus ochrogaster', u'pvole', u'm_ochrogaster', u'79684', u'moch'], u'display_name': u'Prairie vole'}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'ASM342692v1', u'name': u'urocitellus_parryii', u'common_name': u'Arctic ground squirrel', u'accession': u'GCA_003426925.1', u'taxon_id': u'9999', u'strain': None, u'groups': [u'rnaseq', u'core'], u'release': 96, u'display_name': u'Arctic ground squirrel', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'display_name': u'Mouse C3H/HeJ', u'name': u'mus_musculus_c3hhej', u'strain': u'C3H/HeJ', u'accession': u'GCA_001632575.1', u'taxon_id': u'10090', u'common_name': u'house mouse', u'groups': [u'funcgen', u'core'], u'release': 96, u'assembly': u'C3H_HeJ_v1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Panu_3.0', u'name': u'papio_anubis', u'common_name': u'Olive baboon', u'accession': u'GCA_000264685.2', u'taxon_id': u'9555', u'strain': None, u'groups': [u'rnaseq', u'funcgen', u'otherfeatures', u'core'], u'release': 96, u'display_name': u'Olive baboon', u'aliases': [u'olive baboon', u'doguera baboon', u'panubis', u'papanu', u'papio cynocephalus anubis', u'papio hamadryas anubis', u'anubis baboon', u'kenya baboon', u'papio hamadryas doguera', u'9555', u'papio anubis', u'papio doguera', u'panu']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Pmarinus_7.0', u'name': u'petromyzon_marinus', u'display_name': u'Lamprey', u'common_name': u'sea lamprey', u'accession': None, u'strain': None, u'groups': [u'core', u'otherfeatures'], u'taxon_id': u'7757', u'release': 96, u'aliases': [u'petmar', u'7757', u'sea lamprey', u'pmar', u'lamprey', u'pmarinus', u'petromyzon marinus']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Bolivian squirrel monkey', u'name': u'saimiri_boliviensis_boliviensis', u'strain': None, u'accession': u'GCA_000235385.1', u'taxon_id': u'39432', u'common_name': u'Bolivian squirrel monkey', u'groups': [u'funcgen', u'core', u'otherfeatures', u'rnaseq'], u'release': 96, u'assembly': u'SaiBol1.0', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Platyfish', u'name': u'xiphophorus_maculatus', u'strain': None, u'accession': u'GCA_002775205.2', u'taxon_id': u'8083', u'common_name': u'southern platyfish', u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'release': 96, u'assembly': u'X_maculatus-5.0-male', u'aliases': [u'xiphophorus maculatus', u'xipmac', u'xmaculatus', u'8083', u'xmac', u'platyfish', u'southern platyfish']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Cse_v1.0', u'name': u'cynoglossus_semilaevis', u'common_name': u'tongue sole', u'accession': u'GCA_000523025.1', u'strain': None, u'groups': [u'core', u'otherfeatures', u'rnaseq'], u'taxon_id': u'244447', u'release': 96, u'aliases': [], u'display_name': u'Tongue sole'}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Drill', u'name': u'mandrillus_leucophaeus', u'strain': None, u'common_name': u'Drill', u'accession': u'GCA_000951045.1', u'release': 96, u'groups': [u'core', u'otherfeatures', u'funcgen'], u'taxon_id': u'9568', u'assembly': u'Mleu.le_1.0', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'aptHaa1', u'name': u'apteryx_haastii', u'taxon_id': u'8823', u'accession': u'GCA_003342985.1', u'release': 96, u'strain': None, u'groups': [u'core', u'rnaseq'], u'common_name': u'Great spotted kiwi', u'display_name': u'Great spotted kiwi', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Great Tit', u'name': u'parus_major', u'strain': None, u'common_name': u'Great Tit', u'accession': u'GCA_001522545.2', u'release': 96, u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'taxon_id': u'9157', u'assembly': u'Parus_major1.1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Eluc_V3', u'name': u'esox_lucius', u'common_name': u'northern pike', u'accession': u'GCA_000721915.3', u'strain': None, u'groups': [u'core', u'otherfeatures', u'rnaseq'], u'taxon_id': u'8010', u'display_name': u'Northern pike', u'aliases': [], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Nleu_3.0', u'name': u'nomascus_leucogenys', u'common_name': u'Northern white-cheeked gibbon', u'accession': u'GCA_000146795.3', u'taxon_id': u'61853', u'strain': None, u'groups': [u'variation', u'funcgen', u'otherfeatures', u'core'], u'release': 96, u'display_name': u'Gibbon', u'aliases': [u'northern white-cheeked gibbon', u'nleucogenys', u'gibbon', u'nomascus leucogenys', u'nleu', u'nomleu', u'61853']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Macaca_fascicularis_5.0', u'name': u'macaca_fascicularis', u'common_name': u'Crab-eating macaque', u'accession': u'GCA_000364345.1', u'strain': None, u'groups': [u'rnaseq', u'core', u'otherfeatures', u'funcgen'], u'taxon_id': u'9541', u'release': 96, u'aliases': [], u'display_name': u'Crab-eating macaque'}, {u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'assembly': u'NOD_ShiLtJ_v1', u'name': u'mus_musculus_nodshiltj', u'common_name': u'house mouse', u'accession': u'GCA_001624675.1', u'strain': u'NOD/ShiLtJ', u'groups': [u'core', u'funcgen'], u'taxon_id': u'10090', u'release': 96, u'aliases': [], u'display_name': u'Mouse NOD/ShiLtJ'}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Alpine marmot', u'name': u'marmota_marmota_marmota', u'strain': None, u'common_name': u'Alpine marmot', u'accession': u'GCA_001458135.1', u'release': 96, u'groups': [u'rnaseq', u'core', u'otherfeatures'], u'taxon_id': u'9994', u'assembly': u'marMar2.1', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Sloth', u'name': u'choloepus_hoffmanni', u'strain': None, u'accession': None, u'taxon_id': u'9358', u'release': 96, u'groups': [u'core'], u'common_name': u\"Hoffmann's two-fingered sloth\", u'assembly': u'choHof1', u'aliases': [u'choloepus', u'two-toed sloth', u'choloepushoffmanni', u'chof', u\"hoffmann's two-fingered sloth\", u'9358', u'choffmanni', u'chohof', u'choloepus hoffmanni', u'sloth']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'MusPutFur1.0', u'name': u'mustela_putorius_furo', u'taxon_id': u'9669', u'accession': u'GCA_000215625.1', u'release': 96, u'strain': None, u'groups': [u'rnaseq', u'core', u'otherfeatures'], u'common_name': u'Domestic ferret', u'display_name': u'Ferret', u'aliases': [u'mputorius_furo', u'9669', u'mustela putorius furo', u'musput', u'mput', u'domestic ferret', u'ferret']}, {u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'assembly': u'BALB_cJ_v1', u'name': u'mus_musculus_balbcj', u'common_name': u'house mouse', u'accession': u'GCA_001632525.1', u'taxon_id': u'10090', u'strain': u'BALB/cJ', u'groups': [u'core', u'funcgen'], u'release': 96, u'display_name': u'Mouse BALB/cJ', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Chimpanzee', u'name': u'pan_troglodytes', u'strain': None, u'accession': u'GCA_000001515.5', u'taxon_id': u'9598', u'common_name': u'chimpanzee', u'groups': [u'variation', u'rnaseq', u'funcgen', u'core', u'otherfeatures'], u'release': 96, u'assembly': u'Pan_tro_3.0', u'aliases': [u'chimp', u'ptroglodytes', u'ptro', u'pan troglodytes', u'9598', u'pantro', u'common chimpanzee', u'chimpanzee']}, {u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'assembly': u'DBA_2J_v1', u'name': u'mus_musculus_dba2j', u'taxon_id': u'10090', u'accession': u'GCA_001624505.1', u'release': 96, u'strain': u'DBA/2J', u'groups': [u'core', u'funcgen'], u'common_name': u'house mouse', u'display_name': u'Mouse DBA/2J', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'assembly': u'LP_J_v1', u'name': u'mus_musculus_lpj', u'common_name': u'house mouse', u'accession': u'GCA_001632615.1', u'taxon_id': u'10090', u'strain': u'LP/J', u'groups': [u'core', u'funcgen'], u'release': 96, u'display_name': u'Mouse LP/J', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'KH', u'name': u'ciona_intestinalis', u'display_name': u'C.intestinalis', u'common_name': u'Sea squirt Ciona intestinalis', u'accession': u'GCA_000224145.1', u'strain': None, u'groups': [u'otherfeatures', u'core', u'funcgen'], u'taxon_id': u'7719', u'release': 96, u'aliases': [u'sea squirt ciona intestinalis', u'ciona intestinalis', u'cintestinalis', u'c.intestinalis', u'cioint', u'7719', u'cint', u'ciona']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'M_albus_1.0', u'name': u'monopterus_albus', u'common_name': u'swamp eel', u'accession': u'GCA_001952655.1', u'taxon_id': u'43700', u'strain': None, u'groups': [u'core', u'otherfeatures', u'rnaseq'], u'release': 96, u'display_name': u'Swamp eel', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Northern American deer mouse', u'name': u'peromyscus_maniculatus_bairdii', u'strain': None, u'accession': u'GCA_003704035.1', u'taxon_id': u'230844', u'release': 96, u'groups': [u'core'], u'common_name': u'Northern American deer mouse', u'assembly': u'HU_Pman_2.1', u'aliases': [u'northern american deer mouse', u'pmaniculatus_bairdii', u'perman', u'pman', u'230844', u'peromyscus maniculatus bairdii']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'PelSin_1.0', u'name': u'pelodiscus_sinensis', u'common_name': u'Chinese softshell turtle', u'accession': u'GCA_000230535.1', u'strain': None, u'groups': [u'rnaseq', u'otherfeatures', u'core'], u'taxon_id': u'13735', u'display_name': u'Chinese softshell turtle', u'aliases': [u'pelodiscus sinensis', u'13735', u'psinensis', u'chinese softshell turtle', u'p_sinensis', u'chinese soft shell turtle', u'softshell turtle'], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'PPYG2', u'name': u'pongo_abelii', u'common_name': u'Bornean orangutan', u'accession': None, u'taxon_id': u'9601', u'strain': None, u'groups': [u'rnaseq', u'variation', u'otherfeatures', u'core'], u'release': 96, u'display_name': u'Orangutan', u'aliases': [u'orang', u'pongo pygmaeus', u'ppyg', u'9601', u'bornean orangutan', u'pabelii', u'orang_utan', u'pongo_pygmaeus', u'ppygmaeus', u'sumatran orangutan', u'pongo abelii', u'orangutan', u'ponpyg', u'ponabe', u'pabe']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'HLtupMer3', u'name': u'salvator_merianae', u'display_name': u'Argentine black and white tegu', u'common_name': u'Argentine black and white tegu', u'accession': u'GCA_003586115.1', u'strain': None, u'groups': [u'rnaseq', u'core'], u'taxon_id': u'96440', u'release': 96, u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'assembly': u'Caty_1.0', u'name': u'cercocebus_atys', u'common_name': u'Sooty mangabey', u'accession': u'GCA_000955945.1', u'taxon_id': u'9531', u'strain': None, u'groups': [u'funcgen', u'otherfeatures', u'core', u'rnaseq'], u'release': 96, u'display_name': u'Sooty mangabey', u'aliases': []}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Budgerigar', u'name': u'melopsittacus_undulatus', u'strain': None, u'accession': u'GCA_000238935.1', u'common_name': u'Budgie', u'groups': [u'rnaseq', u'core', u'otherfeatures'], u'taxon_id': u'13146', u'assembly': u'Melopsittacus_undulatus_6.3', u'aliases': [], u'release': 96}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Lesser Egyptian jerboa', u'name': u'jaculus_jaculus', u'strain': None, u'common_name': u'Lesser Egyptian jerboa', u'accession': u'GCA_000280705.1', u'release': 96, u'groups': [u'otherfeatures', u'core'], u'taxon_id': u'51337', u'assembly': u'JacJac1.0', u'aliases': [u'jjac', u'jaculus jaculus', u'jjaculus', u'51337', u'jacjac', u'lesser egyptian jerboa']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Hagfish', u'name': u'eptatretus_burgeri', u'strain': None, u'accession': u'GCA_900186335.2', u'taxon_id': u'7764', u'common_name': u'Inshore hagfish', u'groups': [u'rnaseq', u'core'], u'release': 96, u'assembly': u'Eburgeri_3.2', u'aliases': [u'hagfish']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Mexican tetra', u'name': u'astyanax_mexicanus', u'strain': None, u'common_name': u'Mexican tetra', u'accession': u'GCA_000372685.2', u'release': 96, u'groups': [u'rnaseq', u'core', u'otherfeatures'], u'taxon_id': u'7994', u'assembly': u'Astyanax_mexicanus-2.0', u'aliases': [u'amexicanus', u'astmex', u'7994', u'amex', u'astyanax mexicanus', u'cave fish']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Shrew', u'name': u'sorex_araneus', u'strain': None, u'common_name': u'European shrew', u'accession': None, u'release': 96, u'groups': [u'core'], u'taxon_id': u'42254', u'assembly': u'COMMON_SHREW1', u'aliases': [u'sorara', u'european shrew', u'42254', u'shrew', u'ground_shrew', u'saraneus', u'european_shrew', u'sorex araneus', u'sara']}, {u'division': u'EnsemblVertebrates', u'strain_collection': None, u'display_name': u'Yellowtail amberjack', u'name': u'seriola_lalandi_dorsalis', u'strain': None, u'accession': u'GCA_002814215.1', u'taxon_id': u'1841481', u'release': 96, u'groups': [u'otherfeatures', u'core', u'rnaseq'], u'common_name': u'yellowtail amberjack', u'assembly': u'Sedor1', u'aliases': []}]\n"
     ]
    }
   ],
   "source": [
    "print(allsp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<type 'dict'>\n"
     ]
    }
   ],
   "source": [
    "firstsp = allsp[0]\n",
    "print(type(firstsp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{u'division': u'EnsemblVertebrates', u'strain_collection': u'mouse', u'display_name': u'Mouse NZO/HlLtJ', u'name': u'mus_musculus_nzohlltj', u'strain': u'NZO/HlLtJ', u'common_name': u'house mouse', u'accession': u'GCA_001624745.1', u'release': 96, u'groups': [u'funcgen', u'core'], u'taxon_id': u'10090', u'assembly': u'NZO_HlLtJ_v1', u'aliases': []}\n"
     ]
    }
   ],
   "source": [
    "print(firstsp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VB Community Symbol\n",
      "VB External Description\n",
      "VB RNA Description\n",
      "Ribosomal Protein Gene DB\n",
      "ImmunoDB\n",
      "Celera gene\n",
      "Celera peptide\n",
      "Celera transcript\n",
      "European Nucleotide Archive\n",
      "Predicted European Nucleotide Archive\n",
      "FlyBase gene\n",
      "FlyBase translation\n",
      "FlyBase annotation\n",
      "FlyBase symbol\n",
      "FlyBase transcript\n",
      "Gadfly translation CGID\n",
      "FlyBase gene name\n",
      "FlyBase translation name\n",
      "FlyBase transcript name\n",
      "FlyBase gene CGID\n",
      "FlyBase transcript CGID\n",
      "FlyBase translation CGID\n",
      "BioGRID Interaction data, The General Repository for Interaction Datasets\n",
      "GenomeRNAi, a database for cell-based RNAi phenotypes\n",
      "The Interactive Fly\n",
      "Nuclear genes encoding proteins targeted to the mitochondrion\n",
      "FlyReactome, a Curated Knowledgebase of Drosophila Melanogaster Pathways\n",
      "GO\n",
      "GO Reference Collection\n",
      "HGNC Symbol\n",
      "InterPro\n",
      "NCBI gene\n",
      "MGI Symbol\n",
      "MIM gene\n",
      "MIM morbid\n",
      "PDB\n",
      "INSDC protein ID\n",
      "Predicted INSDC Protein ID\n",
      "RefSeq DNA\n",
      "RefSeq mRNA\n",
      "RefSeq ncRNA\n",
      "RefSeq DNA predicted\n",
      "RefSeq mRNA predicted\n",
      "RefSeq ncRNA predicted\n",
      "RefSeq peptide\n",
      "RefSeq peptide predicted\n",
      "RefSeq RNA\n",
      "RefSeq RNA predicted\n",
      "RefSeq Genomic\n",
      "UniProtKB/TrEMBL\n",
      "UniProtKB/Swiss-Prot\n",
      "UniProtKB/SpliceVariant\n",
      "UniProtKB generic accession number (TrEMBL or SwissProt not differentiated)\n",
      "Vega gene\n",
      "Vega transcript\n",
      "Vega translation\n",
      "WormBase Gene\n",
      "WormBase Transcript\n",
      "Wormpep ID\n",
      "WormBase Locus\n",
      "ZFIN\n",
      "ZFIN expression patterns\n",
      "GeneDB\n",
      "JGI Gene ID (JGI 1.0)\n",
      "JGI Gene ID (JGI 2.0)\n",
      "ANISEED (JGI 1.0)\n",
      "ANISEED (JGI 2.0)\n",
      "Ensembl Human Gene\n",
      "Ensembl Human Transcript\n",
      "Ensembl Human Translation\n",
      "RGD Symbol\n",
      "Genoscope pred gene\n",
      "Genoscope pred transcript\n",
      "Genoscope annotated gene\n",
      "Genoscope pred translation\n",
      "Genoscope annotated transcript\n",
      "Genoscope annotated translation\n",
      "miRBase\n",
      "SGD\n",
      "IPI\n",
      "CCDS\n",
      "Sequence Publications\n",
      "UniGene\n",
      "RFAM\n",
      "Xenbase\n",
      "Havana gene\n",
      "Havana transcript\n",
      "Havana translation\n",
      "Havana transcript having same CDS\n",
      "Transcript having exact match between ENSEMBL and HAVANA\n",
      "Ensembl gene\n",
      "Medaka Genome Project\n",
      "Ensembl transcript having exact match with Havana\n",
      "Ensembl transcript sharing CDS with Havana\n",
      "Ensembl Mouse Transcript\n",
      "Ensembl Mouse Translation\n",
      "IMGT/HLA\n",
      "Oxford Ponting Group Monodelphis predictions transcript id\n",
      "Oxford Ponting Group Platypus predictions gene id\n",
      "Oxford Ponting Group Platypus predictions transcript id\n",
      "Ensembl Dog Translation\n",
      "Ensembl Zebrafish Transcript\n",
      "Ensembl Zebrafish Translation\n",
      "Ensembl Chicken Gene\n",
      "Ensembl Chicken Transcript\n",
      "Ensembl Chicken Translation\n",
      "IMCB Fugu Gene\n",
      "Ensembl Fugu Transcript\n",
      "Ensembl Stickleback Translation\n",
      "Ixodes ManualAnnotation\n",
      "VB Community Annotation\n",
      "IMGT/LIGM-DB\n",
      "IMGT/GENE-DB\n",
      "Ensembl Turkey Transcript\n",
      "Ensembl Zebrafinch Transcript\n",
      "Ensembl Cat Translation\n",
      "Platypus olfactory receptor gene\n",
      "PRF\n",
      "PIR\n",
      "Genoscope tetraodon cDNA\n",
      "Stickleback cDNA\n",
      "Platypus 454 cDNA\n",
      "Takifugu cDNA\n",
      "Ciona intestinalis EST\n",
      "Ciona intestinalis cDNA\n",
      "Takifugu EST\n",
      "Genoscope mouse exofish econtig\n",
      "Genoscope fugu exofish econtig\n",
      "Genoscope human exofish econtig\n",
      "Genoscope rat exofish econtig\n",
      "Genoscope chicken exofish econtig\n",
      "Medaka EST\n",
      "Trace Archive id\n",
      "Ciona savignyi EST\n",
      "Generic exdb for WormBase id of any type for feature table\n",
      "Eukaryotic Promoter Database (Bucher)\n",
      "MEROPS - the Peptidase Database\n",
      "TransFac, database of transcription factors and their binding sites\n",
      "UCSC Stable ID\n",
      "Clone-based (Vega) gene\n",
      "Clone-based (Ensembl) gene\n",
      "Clone-based (Vega) transcript\n",
      "Clone-based (Ensembl) transcript\n",
      "DataBase of Aberrant 3' Splice Sites\n",
      "DataBase of Aberrant 5' Splice Sites\n",
      "Human Protein Atlas\n",
      "WikiGene\n",
      "Tgut symbol - Prof. David Burt, Roslin Institute and Royal (Dick) School of Veterinary Studies, Edinburgh University, UK\n",
      "Duck consortium\n",
      "Beijing Genomics Institute (BGI) duck transcriptome\n",
      "GOSlim GOA\n",
      "Wellcome Trust Sanger Institute (WTSI) gorilla transcriptome\n",
      "UniParc\n",
      "BRENDA\n",
      "BioCyc\n",
      "EchoBASE\n",
      "EcoGene\n",
      "SubtiList\n",
      "TubercuList\n",
      "Expression Atlas\n",
      "DIP\n",
      "DisProt\n",
      "DrugBank\n",
      "HAMAP\n",
      "HSSP\n",
      "IntAct\n",
      "KEGG\n",
      "PDBsum\n",
      "PeroxiBase\n",
      "REBASE\n",
      "Reactome\n",
      "RegulonDb\n",
      "EnzymeCommission\n",
      "European Nucleotide Archive\n",
      "JGI ID (JGI 2.0)\n",
      "JGI Gene ID (JGI 2.0 unmapped sequence)\n",
      "JGI Gene ID (JGI 2.0)\n",
      "JGI ID (JGI 2.0 unmapped sequence)\n",
      "JGI ID (JGI 2.0)\n",
      "CADRE\n",
      "RNAmmer\n",
      "tRNAScan-SE\n",
      "DictyBase\n",
      "AspGD\n",
      "AspGD\n",
      "AspGD\n",
      "CADRE\n",
      "CADRE\n",
      "CADRE\n",
      "SGD gene name\n",
      "SGD transcript name\n",
      "JGI ID (JGI 1.1)\n",
      "JGI ID (JGI 1.1)\n",
      "Broad Institute\n",
      "JGI Gene ID (JGI Triad 1.0)\n",
      "JGI Transcript ID (JGI Triad 1.0)\n",
      "JGI ID (JGI Triad 1.0)\n",
      "SchistoDB\n",
      "Broad Institute\n",
      "Broad Institute\n",
      "Broad Institute\n",
      "Broad Institute\n",
      "Broad Institute\n",
      "Broad Institute\n",
      "Broad Institute\n",
      "Pythium genome database\n",
      "Broad Institute\n",
      "Broad Institute\n",
      "Broad Institute\n",
      "Broad Institute\n",
      "Broad Institute\n",
      "Broad Institute\n",
      "SpBase\n",
      "Nectria JGI gene\n",
      "Nectria JGI transcript\n",
      "Nectria JGI\n",
      "Tuber Genome Browser\n",
      "Mycosphaerella JGI gene\n",
      "Broad Institute\n",
      "PomBase\n",
      "PomBase (peptide)\n",
      "CBS Server\n",
      "Cluster of Orthologous Groups (NCBI)\n",
      "Candida Genome Database\n",
      "ProSite ID\n",
      "SMART ID\n",
      "SPD ID\n",
      "PomBase Ontology ID\n",
      "Fission Yeast Phenotype Ontology\n",
      "PSI-MOD, Post Translational Modification Ontology\n",
      "Ashbya Genome Database\n",
      "Ashbya Genome Database\n",
      "EST Cluster (BGI Indica)\n",
      "Enzyme EC Number\n",
      "Flanking Sequence Tag (FST)\n",
      "Gramene\n",
      "GenBank mRNA (Gramene)\n",
      "GenBank EST (Gramene)\n",
      "NASC Gene ID\n",
      "PlantGDB Transcript\n",
      "IRGSP/RAP Gene\n",
      "MSU/TIGR Locus\n",
      "TAIR\n",
      "TAIR Locus (Model)\n",
      "Gene Index\n",
      "Plant Trait Ontology\n",
      "Plant Structure Ontology\n",
      "Gene Ontology\n",
      "Plant Growth Stage\n",
      "Locus Reference Genomic\n",
      "LRG display in Ensembl gene\n",
      "LRG display in Ensembl transcript\n",
      "miRBase transcript name\n",
      "RFAM transcript name\n",
      "HGNC transcript name\n",
      "MGI transcript name\n",
      "ZFIN transcript name\n",
      "Illumina Bodymap transcriptome\n",
      "UniProtKB Gene Name\n",
      "Gibbon Genome Sequencing Consortium\n",
      "Broad Institute of MIT and Harvard\n",
      "PFAM\n",
      "Swine Genome Sequencing Consortium\n",
      "Riken Institute\n",
      "Tasmanian Devil Sequencing Consortium\n",
      "PFAM\n",
      "PFAM\n",
      "PFAM\n",
      "PFAM\n",
      "PFAM\n",
      "GRC primary assembly\n",
      "Ensembl Stickleback Gene\n",
      "Ensembl Stickleback Transcript\n",
      "Chimpanzee Sequencing and Analysis Consortium\n",
      "PomBase Interaction Genetic\n",
      "PomBase Interaction Physical\n",
      "PomBase Gene Name\n",
      "PomBase Systematic ID\n",
      "Orthologous Gene\n",
      "Ensembl Pig Translation\n",
      "Pelodiscus sinensis sequencing consortium\n",
      "Magnaporthe comparative DB\n",
      "Magnaporthe comparative DB\n",
      "Magnaporthe comparative DB\n",
      "Beijing Genomics Institute, Shenzhen\n",
      "International Cat Genome Sequencing Consortium\n",
      "The Genome Institute, Washington University School of Medicine\n",
      "Fly cDNA\n",
      "Fly gold cDNA\n",
      "Ensembl Zebrafinch Translation\n",
      "ToxoDB\n",
      "Sequence Ontology\n",
      "BrassEnsembl\n",
      "BrassEnsembl\n",
      "GOA\n",
      "Quick Go\n",
      "IGDBCAS v1.4 Gene\n",
      "IGDBCAS v1.4 Transcript\n",
      "JGI Gene\n",
      "JGI Transcript\n",
      "IRGSP v1 Gene\n",
      "IRGSP v1 Transcript\n",
      "Ensembl Coelacanth Gene\n",
      "Rat Genome Sequencing Consortium\n",
      "BROAD Institute coelacanth transcriptome\n",
      "Ensembl Anole Lizard Translation\n",
      "BGI Gene\n",
      "GeneIndex\n",
      "Environment Ontology\n",
      "Uppsala University\n",
      "GenBank\n",
      "RefSeq gene name\n",
      "Sol Genomics Network\n",
      "GRC alignment import\n",
      "Broad Institute transcriptome data for monodelphis\n",
      "Swiss Institute of Bioinformatics transcriptome data for monodelphis\n",
      "Swiss Institute of Bioinformatics transcriptome data for platypus\n",
      "Swiss Institute of Bioinformatics transcriptome data for orangutan\n",
      "NCBI species\n",
      "PHI-base accession\n",
      "AnoXcel\n",
      "PGSC\n",
      "UniProt transcript name\n",
      "GMGC\n",
      "TAMU VIBS\n",
      "International Nucleotide Sequence Database Collaboration\n",
      "International Sheep Genome Consortium\n",
      "TAIR Gene Name\n",
      "TIGR Internal ID\n",
      "UniProtKB-Gene Ontology Annotation\n",
      "European Nucleotide Archive\n",
      "European Nucleotide Archive\n",
      "LGBPratt_EST_Cluster\n",
      "Baylor College of Medicine, Texas Medical Center\n",
      "Rhea\n",
      "Ensembl Cow Translation\n",
      "European Nucleotide Archive feature (gene source)\n",
      "European Nucleotide Archive feature (transcript source)\n",
      "European Nucleotide Archive feature (translation source)\n",
      "Anolis Genome Project\n",
      "Turkey Genome Consortium\n",
      "Yutaka Satou Kyoto University\n",
      "International Chicken Genome Consortium\n",
      "TAIR Translation identifier\n",
      "AGI Gene\n",
      "AGI Transcript\n",
      "CGNC Symbol\n",
      "MetaCyc\n",
      "PHI-base phenotype ontology\n",
      "PHI-base condition ontology\n",
      "PubMedCentral\n",
      "PFAM\n",
      "EntrezGene transcript name\n",
      "Broad Institute\n",
      "Broad Institute\n",
      "Broad Institute\n",
      "Ensembl Plants\n",
      "ChEMBL\n",
      "IBSC Gene ID\n",
      "PRIDE\n",
      "LegioList\n",
      "BindingDB\n",
      "STRING\n",
      "UniPathway\n",
      "SWISS-MODEL\n",
      "CYGD\n",
      "Pseudomonas Genome Database\n",
      "MINT\n",
      "Ant Genomes Portal\n",
      "Ensembl Turkey Translation\n",
      "LNCC\n",
      "RGD transcript name\n",
      "Xenbase transcript name\n",
      "NasoniaBase\n",
      "WormBase Gene Sequence-name\n",
      "Sanger Wellcome Trust Institute\n",
      "Ensembl translation having exact match with Havana\n",
      "Ensembl Metazoa\n",
      "Ensembl Macaque Translation\n",
      "PomBase GO AnnotationExtensions\n",
      "UniProtKB/SwissProt Subcellular Locatisation\n",
      "Protein Ontology\n",
      "PomBase PubMed ID\n",
      "Cell Type Ontology\n",
      "ChEBI Chemical Database\n",
      "FYPO Extensions\n",
      "EcoCyc\n",
      "Phenotypic Qualities Ontology\n",
      "UniProtKB Keywords\n",
      "PomBase Quiescence Ontology\n",
      "University of Texas\n",
      "KEGG Pathway and Enzyme\n",
      "Glanville Fritillary Butterfly Genome Project\n",
      "Ensembl internal synonym\n",
      "Digital Object Identifier\n",
      "Zootermopsis nevadensis Genome Project\n",
      "FGDB\n",
      "Ensembl Fungi\n",
      "RNAcentral\n",
      "Strigamia maritima Community Annotation\n",
      "Rfam Gene Name\n",
      "Wheat survey sequence PGSB gene identifier\n",
      "GDEC 3B gene identifier\n",
      "Araport\n",
      "MTGD\n",
      "Reactome gene\n",
      "Reactome transcript\n",
      "PomBase Reference\n",
      "Gene Expression Omnibus\n",
      "Aegilops tauschii GigaDB identifier\n",
      "RegulonDb\n",
      "Experimental Factor Ontology\n",
      "EpiRR\n",
      "VGNC Symbol\n",
      "VGNC transcript name\n",
      "dbSNP\n",
      "VectorBase Proteome\n",
      "Ensembl Protists\n",
      "Mouse Genomes Project\n",
      "Nonhuman Primate Reference Transcriptome Resource\n",
      "Database of genes from NCBI RefSeq genomes\n",
      "NCBI Reference Sequences\n",
      "Ensembl Plants Archive\n",
      "Ensembl Plants\n",
      "MaizeGDB\n",
      "Botrytis community symbol\n",
      "BARLEX: The Barley Genome Explorer \n",
      "PMID 23153364\n",
      "Plant Reactome Reaction ID\n",
      "Plant Reactome Pathway ID\n",
      "Committee on Gene Symbolization, Nomenclature and Linkage of the Rice Genetics Cooperative\n",
      "Integrated Rice Science Database\n",
      "Rice Annotation Project Database\n",
      "Gene names assigned by Gasser Lab\n",
      "Related identifier from previous versions of S. mansoni annotation\n",
      "WormBase CDS\n",
      "BGD\n",
      "ENCODE\n"
     ]
    }
   ],
   "source": [
    "import requests, sys\n",
    " \n",
    "server = \"http://rest.ensembl.org\"\n",
    "ext = \"/info/external_dbs/homo_sapiens?\"\n",
    " \n",
    "r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"})\n",
    " \n",
    "if not r.ok:\n",
    "  r.raise_for_status()\n",
    "  sys.exit()\n",
    " \n",
    "decoded = r.json()\n",
    "#print(type(decoded))\n",
    "#print(decoded[10])\n",
    "\n",
    "for i in decoded:\n",
    "    print(i[u'display_name'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "you are aiming to get 3 species:\n",
    "Rice\n",
    "Wheat\n",
    "Arabidopsis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Findings\n",
    "\n",
    " - unable to extract all the species from ensembl/info/species?\n",
    " - Only got invertebrate and vertebrate animalia names\n",
    " - Rice, wheat and arabidopsis could be in external database, particularly ensembl plants."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ensembl, all assemblies for rice"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<type 'dict'>\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IRGSP-1.0\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IRGSP-1.0\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IRGSP-1.0\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IRGSP-1.0\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IRGSP-1.0\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IRGSP-1.0\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IRGSP-1.0\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IRGSP-1.0\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IRGSP-1.0\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IRGSP-1.0\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IRGSP-1.0\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IRGSP-1.0\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IRGSP-1.0\n"
     ]
    }
   ],
   "source": [
    "import requests, sys\n",
    " \n",
    "server = \"http://rest.ensembl.org\"\n",
    "ext = \"/info/assembly/oryza_sativa?\"\n",
    " \n",
    "r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"})\n",
    " \n",
    "if not r.ok:\n",
    "  r.raise_for_status()\n",
    "  sys.exit()\n",
    " \n",
    "decoded = r.json()\n",
    "print(type(decoded))\n",
    "for key in decoded:\n",
    "    #print(key)\n",
    "    print(\"\\n\")\n",
    "    print(\"Here are names of all assembly\")\n",
    "    print(decoded[u'assembly_name'])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "All the assemblies for rice seem to have come from IRGSP"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ensembl, all assemblies for wheat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<type 'dict'>\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IWGSC\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IWGSC\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IWGSC\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IWGSC\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IWGSC\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IWGSC\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IWGSC\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IWGSC\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IWGSC\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IWGSC\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IWGSC\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IWGSC\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "IWGSC\n"
     ]
    }
   ],
   "source": [
    "import requests, sys\n",
    " \n",
    "server = \"http://rest.ensembl.org\"\n",
    "ext = \"/info/assembly/triticum_aestivum?\"\n",
    " \n",
    "r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"})\n",
    " \n",
    "if not r.ok:\n",
    "  r.raise_for_status()\n",
    "  sys.exit()\n",
    " \n",
    "decoded = r.json()\n",
    "print(type(decoded))\n",
    "for key in decoded:\n",
    "    #print(key)\n",
    "    print(\"\\n\")\n",
    "    print(\"Here are names of all assembly\")\n",
    "    print(decoded[u'assembly_name'])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ensembl, all assembly names for Arabidopsis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<type 'dict'>\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "TAIR10\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "TAIR10\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "TAIR10\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "TAIR10\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "TAIR10\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "TAIR10\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "TAIR10\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "TAIR10\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "TAIR10\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "TAIR10\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "TAIR10\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "TAIR10\n",
      "\n",
      "\n",
      "Here are names of all assembly\n",
      "TAIR10\n"
     ]
    }
   ],
   "source": [
    "import requests, sys\n",
    " \n",
    "server = \"http://rest.ensembl.org\"\n",
    "ext = \"/info/assembly/arabidopsis_thaliana?\"\n",
    " \n",
    "r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"})\n",
    " \n",
    "if not r.ok:\n",
    "  r.raise_for_status()\n",
    "  sys.exit()\n",
    " \n",
    "decoded = r.json()\n",
    "print(type(decoded))\n",
    "for key in decoded:\n",
    "    #print(key)\n",
    "    print(\"\\n\")\n",
    "    print(\"Here are names of all assembly\")\n",
    "    print(decoded[u'assembly_name'])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Conclusion 1\n",
    "\n",
    " - Cannot produce such a list of ensembl species names.\n",
    " - This is because not all databases might be from ensembl. Some might be from NCBI e.g. rice.\n",
    " \n",
    " However,\n",
    " - oryza_sativa works for rice in url\n",
    " - triticum_aestivum works for wheat in url\n",
    " - arabidopsis_thaliana works for arabidopsis in url"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "condensed=\"T:\\colin\\BRice\\gwas-gene-discoveryV2\\MLM.DTF.GWAS.Results\\condensed_genes_designation.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "totalDocSize\n",
      "evidenceTable\n",
      "geneCount\n",
      "docSize\n",
      "geneTable\n",
      "gviewer\n",
      "\n",
      "\n",
      "\n",
      "<type 'str'>\n",
      "\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from __future__ import print_function\n",
    "'''Use the genes from the condensed file to search for Knetminer genome api.'''\n",
    "with open(condensed, \"r\") as fs:\n",
    "    genes = []\n",
    "    with open(\"teststuff.txt\", \"w\") as f:\n",
    "        for line in fs:\n",
    "            col = line.split(\" :: \")\n",
    "            genes.append(col[0])\n",
    "        genelist = (\",\").join(genes) #join all iterative elements by ,\n",
    "        #print(genelist)\n",
    "        pheno = [\"coleoptile length\", \"mesocotyl length\", \"root length\", \"seminal root length\", \"Germination rate\", \"Seedling growth\"]\n",
    "        #use str.join() to convert multiple elments in a list into one string.\n",
    "        keyw = \"%20OR%20\".join(\"({})\".format(i.replace(\" \", \"+AND+\")) for i in pheno)\n",
    "        link = \"http://knetminer.rothamsted.ac.uk/riceknet/genome?\"\n",
    "        parameters = {\"keyword\":keyw, \"list\":genelist}\n",
    "        r = requests.get(link, params=parameters)\n",
    "\n",
    "        if not r.ok:\n",
    "            r.raise_for_status()\n",
    "            sys.exit() #check if request is successful.\n",
    "\n",
    "        decoded = r.json()\n",
    "        #print(type(decoded))\n",
    "        for key in decoded:\n",
    "            print(key)\n",
    "        print(\"\\n\" * 2)\n",
    "\n",
    "        genedata = str(decoded[u'geneTable'])\n",
    "        print(type(genedata))\n",
    "\n",
    "        print(\"\\n\" * 3)\n",
    "\n",
    "        print(genedata, file=f)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "extract gene id and score from unicode geneTable"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Discuss this with Keywan, unicode to str makes it extremely difficult to obtain the genes. Maybe it's better to remove genetable output instead."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Keyword test"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "want to end up with "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "pheno = [\"coleoptile length\", \"mesocotyl length\", \"root length\", \"seminal root length\", \"Germination rate\", \"Seedling growth\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "#location of the mock keyword list\n",
    "klist = \"T:\\colin\\BRice\\mock_keyword_list.txt\"\n",
    "disc_genes=\"T:\\colin\\BRice\\gwas-gene-discoveryV2\\MLM.DTF.GWAS.Results\\summary_genes_discovered.txt\"\n",
    "genetab=\"T:\\colin\\BRice\\gwas-gene-discoveryV2\\MLM.DTF.GWAS.Results\\genetable.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "def keywords(klist):\n",
    "    with open(klist, \"r\") as fk:\n",
    "        pheno = []\n",
    "        for line in fk:\n",
    "            pheno.append(line.rstrip())\n",
    "        return pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Os03g0122600,Os03g0122600,Os03g0122600,Os03g0122600,Os03g0122600,Os06g0164900,Os03g0122600,Os06g0166100,Os06g0156700,Os06g0156600\n",
      "['coleoptile length', 'mesocotyl length', 'root length', 'seminal root length', 'Germination rate', 'Seedling growth']\n",
      "(coleoptile+AND+length)%20OR%20(mesocotyl+AND+length)%20OR%20(root+AND+length)%20OR%20(seminal+AND+root+AND+length)%20OR%20(Germination+AND+rate)%20OR%20(Seedling+AND+growth)\n"
     ]
    }
   ],
   "source": [
    "def knetapi(disc_genes, genetab):\n",
    "    '''Use the genes from the summary file to search for Knetminer genome api.'''\n",
    "    with open(disc_genes, \"r\") as fs:\n",
    "        next(fs)\n",
    "        with open(genetab, \"w\") as fj:\n",
    "            genes = []\n",
    "            for line in fs:\n",
    "                col = line.split(\"\\t\")\n",
    "                genes.append(col[0])\n",
    "            genelist = (\",\").join(genes) #join all iterative elements by ,\n",
    "            print(genelist)\n",
    "            keywords(klist)\n",
    "            #use str.join() to convert multiple elments in a list into one string.\n",
    "            keyw = \"%20OR%20\".join(\"({})\".format(i.replace(\" \", \"+AND+\")) for i in pheno)\n",
    "            print(pheno)\n",
    "            print(keyw)\n",
    "            link = \"http://knetminer.rothamsted.ac.uk/riceknet/genome?\"\n",
    "            parameters = {\"keyword\":keyw, \"list\":genelist}\n",
    "            r = requests.get(link, params=parameters)\n",
    "            \n",
    "            #check if request is succesfsul.\n",
    "            if not r.ok:\n",
    "                r.raise_for_status()\n",
    "                sys.exit()\n",
    "            \n",
    "            #printing out genetable to a file.\n",
    "            decoded = str(r.json()[u'geneTable'])\n",
    "            print(decoded, file=fj)        \n",
    "    #end of knetapi\n",
    "knetapi(disc_genes, genetab)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
