{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# New tests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "kewy_list=\"T:\\colin\\BRice\\V2\\gwas-gene-discovery\\mock_keyword_list.txt\"\n",
    "disc_genes=\"T:\\colin\\BRice\\V2\\gwas-gene-discovery\\MLM.DTF.GWAS.Results\\summary_genes_discovered.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "def knetapi0(disc_genes):\n",
    "    with open(kewy_list, \"r\") as fl:\n",
    "        pheno=[]\n",
    "        for line in fl:\n",
    "            pheno.append(line.rstrip())\n",
    "            #all keyword from plain text file into pheno list.\n",
    "        summary=pd.read_csv(disc_genes, sep=\"\\t\")\n",
    "        genelist=list(summary[u'GENE'])\n",
    "        genestr=(\",\").join(genelist)\n",
    "        #all genes in column 0 of summary into one string\n",
    "        return summary.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>CHROMOSOME</th>\n",
       "      <th>SNP_number</th>\n",
       "      <th>SNP_Base_Position</th>\n",
       "      <th>p-value</th>\n",
       "      <th>logP</th>\n",
       "      <th>SNP_effect</th>\n",
       "      <th>allele_info</th>\n",
       "      <th>Gene_description</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Os03g0122600</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1278045</td>\n",
       "      <td>8.740000e-09</td>\n",
       "      <td>8.058489</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>[u'G', u'A']</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Os03g0122600</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1283193</td>\n",
       "      <td>2.140000e-08</td>\n",
       "      <td>7.669586</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>[u'C', u'T']</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Os03g0122600</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1275151</td>\n",
       "      <td>2.980000e-08</td>\n",
       "      <td>7.525784</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>[u'T', u'C']</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Os03g0122600</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>1286390</td>\n",
       "      <td>4.790000e-08</td>\n",
       "      <td>7.319664</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>[u'G', u'A']</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Os03g0122600</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>1300254</td>\n",
       "      <td>5.040000e-08</td>\n",
       "      <td>7.297569</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[u'C', u'T']</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           GENE  CHROMOSOME  SNP_number  SNP_Base_Position       p-value  \\\n",
       "0  Os03g0122600           3           1            1278045  8.740000e-09   \n",
       "1  Os03g0122600           3           2            1283193  2.140000e-08   \n",
       "2  Os03g0122600           3           3            1275151  2.980000e-08   \n",
       "3  Os03g0122600           3           4            1286390  4.790000e-08   \n",
       "4  Os03g0122600           3           5            1300254  5.040000e-08   \n",
       "\n",
       "       logP           SNP_effect   allele_info Gene_description strand  \n",
       "0  8.058489       intron_variant  [u'G', u'A']             None   None  \n",
       "1  7.669586       intron_variant  [u'C', u'T']             None   None  \n",
       "2  7.525784       intron_variant  [u'T', u'C']             None   None  \n",
       "3  7.319664       intron_variant  [u'G', u'A']             None   None  \n",
       "4  7.297569  5_prime_UTR_variant  [u'C', u'T']             None   None  "
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "knetapi0(disc_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "def knetapi(disc_genes):\n",
    "    with open(kewy_list, \"r\") as fl:\n",
    "        pheno=[]\n",
    "        for line in fl:\n",
    "            pheno.append(line.rstrip())\n",
    "            #all keyword from plain text file into pheno list.\n",
    "        summary=pd.read_csv(disc_genes, sep=\"\\t\")\n",
    "        genelist=list(summary[u'GENE'])\n",
    "        genestr=(\",\").join(genelist)\n",
    "        #all genes in column 0 of summary into one string\n",
    "        \n",
    "        #use str.join() to convert multiple elments in a list into one string.\n",
    "        keyw1 = \"%20OR%20\".join(\"({})\".format(i.replace(\" \", \"+AND+\")) for i in pheno)\n",
    "        link=\"http://knetminer.rothamsted.ac.uk/riceknet/genome?\"\n",
    "        parameters={\"keyword\":keyw1, \"list\":genestr}\n",
    "        r=requests.get(link, params=parameters)\n",
    "        \n",
    "        #check if requests is successful\n",
    "        if not r.ok:\n",
    "                r.raise_for_status()\n",
    "                sys.exit()\n",
    "\n",
    "        #extract unicode string of geneTable decoded from json\n",
    "        decoded=r.json()[u'geneTable'].split(\"\\t\")\n",
    "        #remove space or newline at the end\n",
    "        decoded=(decoded)[:-1]\n",
    "        \n",
    "        colnum=9\n",
    "        #tabulate genetable into 9 columns.\n",
    "        genetable=np.array(decoded).reshape(len(decoded)//colnum, colnum)\n",
    "        genetable=pd.DataFrame(genetable[1:,:], columns=genetable[0,:])\n",
    "        \n",
    "        #delete useless columns from genetable.\n",
    "        genetable=genetable.drop([u'ONDEX-ID', u'GENE NAME', u'TAXID', u'USER', u'QTL'], axis=1)\n",
    "        \n",
    "        #rename genetable columns.\n",
    "        genetable.rename (\n",
    "        columns={\n",
    "            \"ACCESSION\":\"Knetminer_genes\",\n",
    "            \"CHRO\":\"Gene_chromosome\",\n",
    "            \"START\":\"Gene_Base_Position\",\n",
    "            \"SCORE\":\"KnetScore\"\n",
    "        }, inplace=True\n",
    "        )\n",
    "        \n",
    "        return genetable\n",
    "        #compare genetable and summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Knetminer_genes</th>\n",
       "      <th>Gene_chromosome</th>\n",
       "      <th>Gene_Base_Position</th>\n",
       "      <th>KnetScore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OS03G0122600</td>\n",
       "      <td>3</td>\n",
       "      <td>1270320</td>\n",
       "      <td>125.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OS06G0156700</td>\n",
       "      <td>6</td>\n",
       "      <td>2898787</td>\n",
       "      <td>10.64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OS06G0156600</td>\n",
       "      <td>6</td>\n",
       "      <td>2893694</td>\n",
       "      <td>10.64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OS06G0164900</td>\n",
       "      <td>6</td>\n",
       "      <td>3286611</td>\n",
       "      <td>5.94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OS06G0166100</td>\n",
       "      <td>6</td>\n",
       "      <td>3322909</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Knetminer_genes Gene_chromosome Gene_Base_Position KnetScore\n",
       "0    OS03G0122600               3            1270320    125.60\n",
       "1    OS06G0156700               6            2898787     10.64\n",
       "2    OS06G0156600               6            2893694     10.64\n",
       "3    OS06G0164900               6            3286611      5.94\n",
       "4    OS06G0166100               6            3322909      0.00"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "knetapi(disc_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Stuck\n",
    "\n",
    " - can iterate through rows. For every row if genes are the same append knetscore.\n",
    " - can iterate through columns. If columns are the same add relevant value."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
