{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Bugs found in script. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1) This gave invalid Syntax errror. Visual Studio has requested gwas_new_file to be a string encased in \"\" but it was already defined as a file object."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas_new_file=open(out_file, 'w')\n",
    "print('{}\\t{}\\t{}\\t{}\\t{}'.format(\"SNPnum\",\"CHR\",\"snpBP\",\"P\",\"logP\"), file=gwas_new_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "2) The script referring to the outputs from gapit as .txt instead of .csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "original script below"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/usr/bin/env/python\n",
    "import os, math, \n",
    "\n",
    "file=\"T:\\colin\\BRice\\GAPIT_output\\GAPIT.MLM.DTF.GWAS.Results.csv\"\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "for file in os.listdir(\".\"):\n",
    "    #The below line specifies gwas output as .txt not .csv as file is.\n",
    "    if file.endswith(\".Results.txt\"):\n",
    "        print(file)\n",
    "        out_file = file.replace(\"Results\",\"Results_formated\")\n",
    "        gwas_new_file = open(out_file, 'w')\n",
    "        print(type(gwas_new_file))\n",
    "        snp = 0\n",
    "        with open(file) as f:\n",
    "            next(f)\n",
    "            print('{}\\t{}\\t{}\\t{}\\t{}'.format(\"SNPnum\",\"CHR\",\"snpBP\",\"P\",\"logP\"), file=gwas_new_file)\n",
    "            for line in f:\n",
    "                snp += 1\n",
    "                col = line.split(\"\\t\")\n",
    "                chro = col[1]\n",
    "                bp = col[2]\n",
    "                pval = float(col[3])\n",
    "                logP = -math.log10(pval)\n",
    "                #print('{}\\t{}\\t{}\\t{}\\t{}'.format(snp,chro,bp,pval,logP))\n",
    "                print('{}\\t{}\\t{}\\t{}\\t{}'.format(snp,chro,bp,pval,logP), file=gwas_new_file)\n",
    "        \n",
    "        gwas_new_file.close()\n",
    "\n",
    "logP_threshold = 6\n",
    "\n",
    "for file in os.listdir(\".\"):\n",
    "                if file.endswith(\".Results_formated.txt\"):\n",
    "                        print(file)\n",
    "                        out_file_gene_threshold = file.replace(\"Results_formated\",\"Results_formated_threshold\")\n",
    "\n",
    "                        thresholdDB_file = open(out_file_gene_threshold, 'w')\n",
    "                        print(\"SNPnum\\tCHR\\tsnpBP\\tP\\tlogP\", file=thresholdDB_file)\n",
    "                        thresholdDB = []\n",
    "                        with open(file) as f: #was gwas_file\n",
    "                                next(f)\n",
    "                                for line in f:\n",
    "                                        col = line.split(\"\\t\")\n",
    "                                        snp = col[0]\n",
    "                                        chr_snp = int(col[1])\n",
    "                                        bp_snp = int(col[2])\n",
    "                                        pval = float(col[3])\n",
    "                                        logP = float(col[4])\n",
    "                                        if logP > logP_threshold:\n",
    "                                            print('{}\\t{}\\t{}\\t{}\\t{}'.format(snp,chr_snp,bp_snp,pval,logP), file=thresholdDB_file)\n",
    "                        thresholdDB_file.close()\n",
    "                                            \n",
    "\n",
    "\n",
    "\n",
    "def findCloseGene(chr_snp, bp_snp):\n",
    "\tgene_file = 'W:\\\\RICE\\\\Os_Nipponbare_IRGSP_1_gene_Loci_and_designation.txt'\n",
    "\tscore_threshold = 1.0\n",
    "\tmax_distance = 1000\n",
    "\twith open(gene_file) as f:\n",
    "\t\tnext(f)\n",
    "\t\tfor line in f:\n",
    "\t\t\tcol = line.split(\"\\t\")\n",
    "\t\t\tacc = col[0]\n",
    "\t\t\tchr_gene = int(col[1])\n",
    "\t\t\tbeg_gene = int(col[2])\n",
    "\t\t\tend_gene = int(col[3])\n",
    "\t\t\tdesign = col[4]\n",
    "\t\t\t#score = float(col[6])\n",
    "\t\t\t#SNP is within gene\n",
    "\t\t\tif (chr_snp == chr_gene) & (bp_snp > beg_gene) & (bp_snp < end_gene):\n",
    "\t\t\t\t#print('Found SNP within gene {}'.format(acc)) \n",
    "\t\t\t\treturn acc\n",
    "\t\t\t#SNP is down stream of gene with < max_distance \n",
    "\t\t\telif (chr_snp == chr_gene) & (abs(bp_snp - beg_gene) < max_distance):\n",
    "\t\t\t\t#print('Found SNP downstream of {}'.format(acc)) \n",
    "\t\t\t\treturn acc\n",
    "\t\t\t#SNP is up stream of gene with < max_distance \n",
    "\t\t\telif (chr_snp == chr_gene) & (abs(bp_snp - end_gene) < max_distance):\n",
    "\t\t\t\t#print('Found SNP upstream of {}'.format(acc)) \n",
    "\t\t\t\treturn acc\n",
    "\treturn \"FooBar\"\n",
    "\n",
    "\n",
    "def findCloseGeneDesign(chr_snp, bp_snp):\n",
    "\tgene_file = 'W:\\\\RICE\\\\Os_Nipponbare_IRGSP_1_gene_Loci_and_designation.txt'\n",
    "\tscore_threshold = 1.0\n",
    "\tmax_distance = 1000\n",
    "\twith open(gene_file) as f:\n",
    "\t\tnext(f)\n",
    "\t\tfor line in f:\n",
    "\t\t\tcol = line.split(\"\\t\")\n",
    "\t\t\tacc = col[0]\n",
    "\t\t\tchr_gene = int(col[1])\n",
    "\t\t\tbeg_gene = int(col[2])\n",
    "\t\t\tend_gene = int(col[3])\n",
    "\t\t\tdesign = col[4]\n",
    "\t\t\t#score = float(col[6])\n",
    "\t\t\t#SNP is within gene\n",
    "\t\t\tif (chr_snp == chr_gene) & (bp_snp > beg_gene) & (bp_snp < end_gene):\n",
    "\t\t\t\t#print('Found SNP within gene {}'.format(design)) \n",
    "\t\t\t\treturn design\n",
    "\t\t\t#SNP is down stream of gene with < max_distance \n",
    "\t\t\telif (chr_snp == chr_gene) & (abs(bp_snp - beg_gene) < max_distance):\n",
    "\t\t\t\t#print('Found SNP downstream of {}'.format(design)) \n",
    "\t\t\t\treturn design\n",
    "\t\t\t#SNP is up stream of gene with < max_distance \n",
    "\t\t\telif (chr_snp == chr_gene) & (abs(bp_snp - end_gene) < max_distance):\n",
    "\t\t\t\t#print('Found SNP upstream of {}'.format(design)) \n",
    "\t\t\t\treturn design\n",
    "\treturn \"FooBar2\"\n",
    "\n",
    "\n",
    "if __name__ == '__main__':\n",
    "\n",
    "\n",
    "        for file in os.listdir(\".\"):\n",
    "                if file.endswith(\".Results_formated_threshold.txt\"):\n",
    "                        print(file)\n",
    "                        out_file_gwas = file.replace(\"Results_formated_threshold\",\"Results_formated_gdp_FINAL\")\n",
    "                        out_file_gene = file.replace(\"Results_formated_threshold\",\"Results_formated_gene\")\n",
    "                        out_file_genedesign = file.replace(\"Results_formated_threshold\",\"Results_formated_gene_and_designation\")\n",
    "                        gwas_new_file = open(out_file_gwas, 'w')\n",
    "                        print(\"GENE\\tCHR\\tSNPnum\\tsnpBP\\tP\\tlogP\\tdesignation\", file=gwas_new_file)\n",
    "                        highlight = []\n",
    "                        highlightdesign = []\n",
    "                        with open(file) as f: #was gwas_file\n",
    "                                next(f)\n",
    "                                for line in f:\n",
    "                                        col = line.split(\"\\t\")\n",
    "                                        snp = col[0]\n",
    "                                        snpnum = col[0]\n",
    "                                        chr_snp = int(col[1])\n",
    "                                        bp_snp = int(col[2])\n",
    "                                        pval = float(col[3])\n",
    "                                        logP = float(col[4])\n",
    "                                        gene = findCloseGene(chr_snp, bp_snp)\n",
    "                                        design = findCloseGeneDesign(chr_snp, bp_snp)\n",
    "                                        blank = \"-\"\n",
    "                                        if gene == \"FooBar\":\n",
    "                                                print('{}\\t{}\\t{}\\t{}\\t{}\\t{}'.format(blank,blank,blank,blank,blank,blank), file=gwas_new_file)\n",
    "                                        else:\n",
    "                                                print('{}\\t{}\\t{}\\t{}\\t{}\\t{}\\t{}'.format(gene,chr_snp,snpnum,bp_snp,pval,logP,design))\n",
    "                                                if logP > logP_threshold:\n",
    "                                                        highlight.append(gene)\n",
    "                                                        highlightdesign.append(design)\n",
    "                                                        print('{}\\t{}\\t{}\\t{}\\t{}\\t{}\\t{}'.format(gene,chr_snp,snpnum,bp_snp,pval,logP,design), file=gwas_new_file)\n",
    "                                                else:\n",
    "                                                        print('{}\\t{}\\t{}\\t{}\\t{}\\t{}'.format(snp,chr_snp,snpnum,bp_snp,pval,logP), file=gwas_new_file)\n",
    "                                                \n",
    "                        gwas_new_file.close()\n",
    "                      \n",
    "                        \n",
    "                        highlight_file = open(out_file_gene, 'w')\n",
    "\n",
    "\n",
    "                        for gene in set(highlight):\n",
    "                            print(gene, file=highlight_file)\n",
    "\n",
    "                        highlight_file.close()\n",
    "                        highlightdesign_file = open(out_file_genedesign, 'w')\n",
    "                        \n",
    "                        for design in set(highlightdesign):\n",
    "                            print(design, file=highlightdesign_file) \n",
    "                                \n",
    "                        highlightdesign_file.close()\n",
    "\n",
    "       \n",
    "             "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
